<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>2. Package Documentation &#8212; modlAMP v2.6.0 documentation</title>
    
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/custom.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     'v2.6.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/modlamp.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="modlAMP v2.6.0 documentation" href="index.html" />
    <link rel="next" title="3. Example scripts" href="examplescript.html" />
    <link rel="prev" title="1. README" href="readme.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="examplescript.html" title="3. Example scripts"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="readme.html" title="1. README"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">modlAMP v2.6.0 documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="package-documentation">
<h1>2. Package Documentation<a class="headerlink" href="#package-documentation" title="Permalink to this headline">¶</a></h1>
<p>The <strong>modlamp</strong> package includes the following <strong>modules</strong>:</p>
<table border="1" class="docutils">
<colgroup>
<col width="20%" />
<col width="80%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Module</th>
<th class="head">Characteristics</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#module-modlamp.descriptors" title="modlamp.descriptors"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.descriptors</span></code></a></td>
<td>Module incorporating <code class="xref py py-class docutils literal"><span class="pre">GlobalDescriptor</span></code> and <code class="xref py py-class docutils literal"><span class="pre">PeptideDescriptor</span></code> for descriptor calculations.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#module-modlamp.sequences" title="modlamp.sequences"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.sequences</span></code></a></td>
<td>Module incorporating different classes for generating peptide sequences with different characteristics.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#module-modlamp.database" title="modlamp.database"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.database</span></code></a></td>
<td>Module incorporating functions to connect to the modlab in-house peptide database server.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#module-modlamp.datasets" title="modlamp.datasets"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.datasets</span></code></a></td>
<td>Module incorporating functions to load different peptide datasets used for ML classification tasks.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#module-modlamp.plot" title="modlamp.plot"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.plot</span></code></a></td>
<td>Module incorporating functions to visualize descriptor.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#module-modlamp.ml" title="modlamp.ml"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.ml</span></code></a></td>
<td>Module incorporating machine learning functions for training SVM or RF models on peptides.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#module-modlamp.wetlab" title="modlamp.wetlab"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.wetlab</span></code></a></td>
<td>Module incorporating classes for analysis of wetlab data like circular dichroism plots.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#module-modlamp.analysis" title="modlamp.analysis"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.analysis</span></code></a></td>
<td>Module incorporating the <code class="xref py py-class docutils literal"><span class="pre">GlobalAnalysis</span></code> class for sequence library analysis and comparison.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#module-modlamp.core" title="modlamp.core"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.core</span></code></a></td>
<td>Module incorporating helper functions for other modules.</td>
</tr>
</tbody>
</table>
<div class="section" id="module-modlamp.descriptors">
<span id="modlamp-descriptors"></span><h2>2.1. modlamp.descriptors<a class="headerlink" href="#module-modlamp.descriptors" title="Permalink to this headline">¶</a></h2>
<p>This module incorporates different classes to calculate peptide descriptor values. The following classes are available:</p>
<table border="1" class="docutils">
<colgroup>
<col width="28%" />
<col width="72%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Class</th>
<th class="head">Characteristics</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#modlamp.descriptors.GlobalDescriptor" title="modlamp.descriptors.GlobalDescriptor"><code class="xref py py-class docutils literal"><span class="pre">GlobalDescriptor</span></code></a></td>
<td>Global one-dimensional peptide descriptors calculated from the AA sequence.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor" title="modlamp.descriptors.PeptideDescriptor"><code class="xref py py-class docutils literal"><span class="pre">PeptideDescriptor</span></code></a></td>
<td>AA scale based global or convoluted descriptors (auto-/cross-correlated).</td>
</tr>
</tbody>
</table>
<dl class="class">
<dt id="modlamp.descriptors.GlobalDescriptor">
<em class="property">class </em><code class="descclassname">modlamp.descriptors.</code><code class="descname">GlobalDescriptor</code><span class="sig-paren">(</span><em>seqs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for global, non-amino acid scale dependant descriptors. The following descriptors can be calculated by
the <strong>methods</strong> linked below:</p>
<ul class="simple">
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.length">Sequence Length</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.calculate_MW">Molecular Weight</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.calculate_charge">Sequence Charge</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.charge_density">Charge Density</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.isoelectric_point">Isoelectric Point</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.instability_index">Instability Index</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.aromaticity">Aromaticity</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.aliphatic_index">Aliphatic Index</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.boman_index">Boman Index</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.hydrophobic_ratio">Hydrophobic Ratio</a></li>
<li><a class="reference external" href="modlamp.html#modlamp.descriptors.GlobalDescriptor.calculate_all">all of the above</a></li>
</ul>
<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>seqs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.__init__" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seqs</strong> &#8211; a .fasta file with sequences, a list of sequences or a single sequence as string to calculate the
descriptor values for.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">initialized lists self.sequences, self.names and dictionary self.AA with amino acid scale values</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.descriptors</span> <span class="k">import</span> <span class="n">GlobalDescriptor</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;KLAKLAKKLAKLAK&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KLAKLAKKLAKLAK&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;/Path/to/file.fasta&#39;</span><span class="p">)</span>  <span class="c1"># load sequences from .fasta file</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;AFDGHLKI&#39;,&#39;KKLQRSDLLRTK&#39;,&#39;KKLASCNNIPPR&#39;...]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.length">
<code class="descname">length</code><span class="sig-paren">(</span><em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.length"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.length" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the length (total AA count) of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">array of sequence lengths in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">([</span><span class="s1">&#39;AFDGHLKI&#39;</span><span class="p">,</span><span class="s1">&#39;KKLQRSDLLRTK&#39;</span><span class="p">,</span><span class="s1">&#39;KKLASCNNIPPR&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">length</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 8], [12], [12]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.calculate_MW">
<code class="descname">calculate_MW</code><span class="sig-paren">(</span><em>amide=False</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.calculate_MW"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.calculate_MW" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the molecular weight [g/mol] of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>amide</strong> &#8211; {boolean} whether the sequences are C-terminally amidated (subtracts 0.95 from the MW).</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;IAESFKGHIPL&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">calculate_MW</span><span class="p">(</span><span class="n">amide</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 1210.43]])</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.aa_weights" title="modlamp.core.aa_weights"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.aa_weights()</span></code></a></p>
</div>
<div class="versionchanged">
<p><span class="versionmodified">Changed in version v2.1.5: </span>amide option added</p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.calculate_charge">
<code class="descname">calculate_charge</code><span class="sig-paren">(</span><em>ph=7.0</em>, <em>amide=False</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.calculate_charge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.calculate_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to overall charge of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.</p>
<p>The method used is first described by Bjellqvist. In the case of amidation, the value for the &#8216;Cterm&#8217; pKa is 15
(and Cterm is added to the pos_pKs dictionary.
The pKa scale is extracted from: <a class="reference external" href="http://www.hbcpnetbase.com/">http://www.hbcpnetbase.com/</a> (CRC Handbook of Chemistry and Physics, 96th ed).</p>
<p><strong>pos_pKs</strong> = {&#8216;Nterm&#8217;: 9.38, &#8216;K&#8217;: 10.67, &#8216;R&#8217;: 12.10, &#8216;H&#8217;: 6.04}</p>
<p><strong>neg_pKs</strong> = {&#8216;Cterm&#8217;: 2.15, &#8216;D&#8217;: 3.71, &#8216;E&#8217;: 4.15, &#8216;C&#8217;: 8.14, &#8216;Y&#8217;: 10.10}</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ph</strong> &#8211; {float} ph at which to calculate peptide charge.</li>
<li><strong>amide</strong> &#8211; {boolean} whether the sequences have an amidated C-terminus.</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;KLAKFGKRSELVALSG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">calculate_charge</span><span class="p">(</span><span class="n">ph</span><span class="o">=</span><span class="mf">7.4</span><span class="p">,</span> <span class="n">amide</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 3.989]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.charge_density">
<code class="descname">charge_density</code><span class="sig-paren">(</span><em>ph=7.0</em>, <em>amide=False</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.charge_density"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.charge_density" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the charge density (charge / MW) of every sequences in the attributes <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ph</strong> &#8211; {float} pH at which to calculate peptide charge.</li>
<li><strong>amide</strong> &#8211; {boolean} whether the sequences have an amidated C-terminus.</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;GNSDLLIEQRTLLASDEF&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">charge_density</span><span class="p">(</span><span class="n">ph</span><span class="o">=</span><span class="mi">6</span><span class="p">,</span> <span class="n">amide</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[-0.00097119]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.isoelectric_point">
<code class="descname">isoelectric_point</code><span class="sig-paren">(</span><em>amide=False</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.isoelectric_point"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.isoelectric_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the isoelectric point of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
The pK scale is extracted from: <a class="reference external" href="http://www.hbcpnetbase.com/">http://www.hbcpnetbase.com/</a> (CRC Handbook of Chemistry and Physics, 96th ed).</p>
<blockquote>
<div><p><strong>pos_pKs</strong> = {&#8216;Nterm&#8217;: 9.38, &#8216;K&#8217;: 10.67, &#8216;R&#8217;: 12.10, &#8216;H&#8217;: 6.04}</p>
<p><strong>neg_pKs</strong> = {&#8216;Cterm&#8217;: 2.15, &#8216;D&#8217;: 3.71, &#8216;E&#8217;: 4.15, &#8216;C&#8217;: 8.14, &#8216;Y&#8217;: 10.10}</p>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>amide</strong> &#8211; {boolean} whether the sequences have an amidated C-terminus.</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;KLFDIKFGHIPQRST&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">isoelectric_point</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 10.6796875]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.instability_index">
<code class="descname">instability_index</code><span class="sig-paren">(</span><em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.instability_index"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.instability_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the instability of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
The instability index is a prediction of protein stability based on the amino acid composition.
([1] K. Guruprasad, B. V Reddy, M. W. Pandit, Protein Eng. 1990, 4, 155–161.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;LLASMNDLLAKRST&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">instability_index</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 63.95714286]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.aromaticity">
<code class="descname">aromaticity</code><span class="sig-paren">(</span><em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.aromaticity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.aromaticity" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the aromaticity of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
According to Lobry, 1994, it is simply the relative frequency of Phe+Trp+Tyr.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;GLFYWRFFLQRRFLYWW&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">aromaticity</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.52941176]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.aliphatic_index">
<code class="descname">aliphatic_index</code><span class="sig-paren">(</span><em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.aliphatic_index"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.aliphatic_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the aliphatic index of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
According to Ikai, 1980, the aliphatic index is a measure of thermal stability of proteins and is dependant
on the relative volume occupied by aliphatic amino acids (A,I,L &amp; V).
([1] A. Ikai, J. Biochem. 1980, 88, 1895–1898.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;KWLKYLKKLAKLVK&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">aliphatic_index</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 139.28571429]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.boman_index">
<code class="descname">boman_index</code><span class="sig-paren">(</span><em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.boman_index"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.boman_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the boman index of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
According to Boman, 2003, the boman index is a measure for protein-protein interactions and is calculated by
summing over all amino acid free energy of transfer [kcal/mol] between water and cyclohexane,[2] followed by
dividing by    sequence length.
([1] H. G. Boman, D. Wade, I. a Boman, B. Wåhlin, R. B. Merrifield, <em>FEBS Lett</em>. <strong>1989</strong>, <em>259</em>, 103–106.
[2] A. Radzick, R. Wolfenden, <em>Biochemistry</em> <strong>1988</strong>, <em>27</em>, 1664–1670.)</p>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.aa_energies" title="modlamp.core.aa_energies"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.aa_energies()</span></code></a></p>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">boman_index</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[-1.011875]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.hydrophobic_ratio">
<code class="descname">hydrophobic_ratio</code><span class="sig-paren">(</span><em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.hydrophobic_ratio"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.hydrophobic_ratio" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the hydrophobic ratio of every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>, which is the
relative frequency of the amino acids <strong>A,C,F,I,L,M &amp; V</strong>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;VALLYWRTVLLAIII&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">hydrophobic_ratio</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.73333333]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.calculate_all">
<code class="descname">calculate_all</code><span class="sig-paren">(</span><em>ph=7.4</em>, <em>amide=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.calculate_all"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.calculate_all" title="Permalink to this definition">¶</a></dt>
<dd><p>Method combining all global descriptors and appending them into the feature matrix in the attribute
<code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ph</strong> &#8211; {float} pH at which to calculate peptide charge</li>
<li><strong>amide</strong> &#8211; {boolean} whether the sequences have an amidated C-terminus.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">array of descriptor values in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">d</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">(</span><span class="s1">&#39;AFGHFKLKKLFIFGHERT&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">calculate_all</span><span class="p">(</span><span class="n">amide</span><span class="o">=</span><span class="n">true</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">featurenames</span>
<span class="go">[&#39;MW&#39;, &#39;ChargeDensity&#39;, &#39;pI&#39;, &#39;InstabilityInd&#39;, &#39;Aromaticity&#39;, &#39;AliphaticInd&#39;, &#39;BomanInd&#39;, &#39;HydrophRatio&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[  2.17559000e+03,   1.87167619e-03,   1.16757812e+01, ...  1.10555556e+00,   4.44444444e-01]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.feature_scaling">
<code class="descname">feature_scaling</code><span class="sig-paren">(</span><em>stype='standard'</em>, <em>fit=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.feature_scaling"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.feature_scaling" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for feature scaling of the calculated descriptor matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>stype</strong> &#8211; {str} <strong>&#8216;standard&#8217;</strong> or <strong>&#8216;minmax&#8217;</strong>, type of scaling to be used</li>
<li><strong>fit</strong> &#8211; {boolean}, defines whether the used scaler is first fitting on the data (True) or
whether the already fitted scaler in <code class="xref py py-attr docutils literal"><span class="pre">scaler</span></code> should be used to transform (False).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">scaled descriptor values in <code class="xref py py-attr docutils literal"><span class="pre">self.descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[0.155],[0.34],[0.16235294],[-0.08842105],[0.116]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">feature_scaling</span><span class="p">(</span><span class="n">stype</span><span class="o">=</span><span class="s1">&#39;minmax&#39;</span><span class="p">,</span><span class="n">fit</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">array([[0.56818182],[1.],[0.5853447],[0.],[0.47714988]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.feature_shuffle">
<code class="descname">feature_shuffle</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.feature_shuffle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.feature_shuffle" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for shuffling features randomly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">descriptor matrix with shuffled feature columns in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[0.80685625,167.05234375,39.56818125,-0.26338667,155.16888667,33.48778]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">feature_shuffle</span><span class="p">()</span>
<span class="go">array([[155.16888667,-0.26338667,167.05234375,0.80685625,39.56818125,33.48778]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.filter_values">
<code class="descname">filter_values</code><span class="sig-paren">(</span><em>values</em>, <em>operator='=='</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.filter_values"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.filter_values" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter the descriptor matrix in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code> for a given list of values (same
size as the number of features in the descriptor matrix!) The operator option tells the method whether to
filter for values equal, lower, higher ect. to the given values in the <strong>values</strong> array.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>values</strong> &#8211; List/array of values to filter the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code> for</li>
<li><strong>operator</strong> &#8211; filter criterion, available are all SQL like operators: <code class="docutils literal"><span class="pre">==</span></code>, <code class="docutils literal"><span class="pre">&lt;</span></code>, <code class="docutils literal"><span class="pre">&gt;</span></code>, <code class="docutils literal"><span class="pre">&lt;=``and</span> <span class="pre">``&gt;=</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">filtered descriptor matrix and updated sequences in the corresponding attributes.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">min</span><span class="p">(</span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">array([ 0.04523998])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(412, 1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">filter_values</span><span class="p">([</span><span class="mf">0.1</span><span class="p">],</span> <span class="n">operator</span><span class="o">=</span><span class="s1">&#39;&gt;&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">min</span><span class="p">(</span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">array([ 0.10344828])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">descriptor</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(168, 1)</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_values" title="modlamp.core.filter_values"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_values()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.filter_duplicates">
<code class="descname">filter_duplicates</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.filter_duplicates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.filter_duplicates" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter duplicates in the sequences from the class attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequences list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">([</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;GLFDIVKKVVGGKKASSERT&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">filter_duplicates</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KLLKLLKKLLKLLK&#39;, &#39;GLFDIVKKVVGGKKASSERT&#39;]</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_sequences" title="modlamp.core.filter_sequences"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_sequences()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.5.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.keep_natural_aa">
<code class="descname">keep_natural_aa</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.keep_natural_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.keep_natural_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences that do not contain natural amino acids. If the sequence contains a character
that is not in [&#8216;A&#8217;,&#8217;C&#8217;,&#8217;D,&#8217;E&#8217;,&#8217;F&#8217;,&#8217;G&#8217;,&#8217;H&#8217;,&#8217;I&#8217;,&#8217;K&#8217;,&#8217;L&#8217;,&#8217;M&#8217;,&#8217;N&#8217;,&#8217;P&#8217;,&#8217;Q&#8217;,&#8217;R&#8217;,&#8217;S&#8217;,&#8217;T&#8217;,&#8217;V&#8217;,&#8217;W&#8217;,&#8217;Y&#8217;].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>. The other attributes are also filtered
accordingly.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">([</span><span class="s1">&#39;ACDEFGHIKLMNPQRSTVWY&#39;</span><span class="p">,</span> <span class="s1">&#39;JOGHURT&#39;</span><span class="p">,</span> <span class="s1">&#39;ELVISISKING&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">keep_natural_aa</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;ACDEFGHIKLMNPQRSTVWY&#39;, &#39;ELVISISKING&#39;]</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.keep_natural_aa" title="modlamp.core.keep_natural_aa"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.keep_natural_aa()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.5.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.filter_aa">
<code class="descname">filter_aa</code><span class="sig-paren">(</span><em>aminoacids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.filter_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.filter_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter sequences and corresponding descriptor values, if the sequences contain any of the given
amino acids in the argument list <strong>aminoacids</strong>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aminoacids</strong> &#8211; List/array of amino acids {str.upper} to filter for</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">filtered descriptor matrix and updated sequences and names in the corresponding attributes.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">([</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;ACCDFACAD&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">filter_aa</span><span class="p">([</span><span class="s1">&#39;C&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KLLKLLKKLLKLLK&#39;]</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_aa" title="modlamp.core.filter_aa"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_aa()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.filter_sequences">
<code class="descname">filter_sequences</code><span class="sig-paren">(</span><em>sequences</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.filter_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.filter_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out entries for given sequences in <em>sequences</em> out of a descriptor instance. All
corresponding fields of these sequences (<em>descriptor</em>, <em>name</em>) are deleted as well. The method returns an
updated descriptor instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>sequences</strong> &#8211; {list} sequences to be filtered out of the whole instance, including corresponding data</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">updated instance</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span> <span class="o">=</span> <span class="n">GlobalDescriptor</span><span class="p">([</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;KLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;KLLLKLKLKLL&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">filter_sequences</span><span class="p">([</span><span class="s1">&#39;KLLK&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">desc</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KLLKLLKKLLKLLK&#39;, &#39;KLLLKLKLKLL&#39;]</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_sequences" title="modlamp.core.filter_sequences"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_sequences()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.4.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.filter_unnatural">
<code class="descname">filter_unnatural</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.filter_unnatural"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.filter_unnatural" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences with unnatural amino acids from <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
:return: Filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></p>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_unnatural" title="modlamp.core.filter_unnatural"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_unnatural()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.random_selection">
<code class="descname">random_selection</code><span class="sig-paren">(</span><em>num</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.random_selection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.random_selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to select a random number of sequences (with names and descriptors if present) out of a given
descriptor instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>num</strong> &#8211; {int} number of entries to be randomly selected</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">updated instance</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.random_selection" title="modlamp.core.random_selection"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.random_selection()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.3.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.minmax_selection">
<code class="descname">minmax_selection</code><span class="sig-paren">(</span><em>iterations</em>, <em>distmetric='euclidean'</em>, <em>randseed=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.minmax_selection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.minmax_selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to select a specified number of sequences out of a given descriptor instance according to the
minmax algorithm.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>iterations</strong> &#8211; {int} number of sequences to retrieve.</li>
<li><strong>distmetric</strong> &#8211; {str} distance metric to calculate the distances between the sequences in descriptor space.
Choose from scipy.spacial.distance (<a class="reference external" href="http://docs.scipy.org/doc/scipy/reference/spatial.distance.html">http://docs.scipy.org/doc/scipy/reference/spatial.distance.html</a>).
E.g. &#8216;euclidean&#8217;, &#8216;minkowsky&#8217;.</li>
<li><strong>randseed</strong> &#8211; {int} Set a random seed for numpy to pick the first sequence.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">updated instance</p>
</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.minmax_selection" title="modlamp.core.minmax_selection"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.minmax_selection()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.6.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.load_descriptordata">
<code class="descname">load_descriptordata</code><span class="sig-paren">(</span><em>filename</em>, <em>delimiter='</em>, <em>'</em>, <em>targets=False</em>, <em>header=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.load_descriptordata"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.load_descriptordata" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to load any data file with sequences and descriptor values and save it to a new insatnce of the
class <a class="reference internal" href="#modlamp.descriptors.GlobalDescriptor" title="modlamp.descriptors.GlobalDescriptor"><code class="xref py py-class docutils literal"><span class="pre">modlamp.descriptors.GlobalDescriptor</span></code></a>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> &#8211; {str} filename of the data file to be loaded</li>
<li><strong>delimiter</strong> &#8211; {str} column delimiter</li>
<li><strong>targets</strong> &#8211; {boolean} whether last column in the file contains a target class vector</li>
<li><strong>header</strong> &#8211; {int} number of header lines to skip in the file</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">loaded sequences, descriptor values and targets in the corresponding attributes.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.save_descriptor">
<code class="descname">save_descriptor</code><span class="sig-paren">(</span><em>filename</em>, <em>delimiter='</em>, <em>'</em>, <em>targets=None</em>, <em>header=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.save_descriptor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.save_descriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to save the descriptor values to a .csv/.txt file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> &#8211; {str} filename of the output file</li>
<li><strong>delimiter</strong> &#8211; {str} column delimiter</li>
<li><strong>targets</strong> &#8211; {list} target class vector to be added to descriptor (same length as <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>)</li>
<li><strong>header</strong> &#8211; {str} header to be written at the beginning of the file</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">output file with peptide names and descriptor values</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.GlobalDescriptor.save_fasta">
<code class="descname">save_fasta</code><span class="sig-paren">(</span><em>outputfile</em>, <em>names=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#GlobalDescriptor.save_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.GlobalDescriptor.save_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for saving sequences from <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> to a FASTA formatted file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>outputfile</strong> &#8211; {str} filename of the output FASTA file</li>
<li><strong>names</strong> &#8211; {boolean} whether sequence names from self.names should be saved as sequence identifiers</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">list of sequences in self.sequences with corresponding sequence names in the attribute <code class="xref py py-attr docutils literal"><span class="pre">names</span></code></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.descriptors.PeptideDescriptor">
<em class="property">class </em><code class="descclassname">modlamp.descriptors.</code><code class="descname">PeptideDescriptor</code><span class="sig-paren">(</span><em>seqs</em>, <em>scalename='eisenberg'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for peptide descriptors. The following <strong>amino acid descriptor scales</strong> are available for descriptor
calculation:</p>
<ul class="simple">
<li><strong>AASI</strong>           (An amino acid selectivity index scale for helical antimicrobial peptides, <em>[1] D. Juretić, D. Vukicević, N. Ilić, N. Antcheva, A. Tossi, J. Chem. Inf. Model. 2009, 49, 2873–2882.</em>)</li>
<li><strong>ABHPRK</strong>          (modlabs inhouse physicochemical feature scale (Acidic, Basic, Hydrophobic, Polar, aRomatic, Kink-inducer)</li>
<li><strong>argos</strong>          (Argos hydrophobicity amino acid scale, <em>[2] Argos, P., Rao, J. K. M. &amp; Hargrave, P. A., Eur. J. Biochem. 2005, 128, 565–575.</em>)</li>
<li><strong>bulkiness</strong>      (Amino acid side chain bulkiness scale, <em>[3] J. M. Zimmerman, N. Eliezer, R. Simha, J. Theor. Biol. 1968, 21, 170–201.</em>)</li>
<li><strong>charge_physio</strong>  (Amino acid charge at pH 7.0 - Hystidine charge +0.1.)</li>
<li><strong>charge_acidic</strong>  (Amino acid charge at acidic pH - Hystidine charge +1.0.)</li>
<li><strong>cougar</strong>         (modlabs inhouse selection of global peptide descriptors)</li>
<li><strong>eisenberg</strong>      (the Eisenberg hydrophobicity consensus amino acid scale, <em>[4] D. Eisenberg, R. M. Weiss, T. C. Terwilliger, W. Wilcox, Faraday Symp. Chem. Soc. 1982, 17, 109.</em>)</li>
<li><strong>Ez</strong>             (potential that assesses energies of insertion of amino acid side chains into lipid bilayers, <em>[5] A. Senes, D. C. Chadi, P. B. Law, R. F. S. Walters, V. Nanda, W. F. DeGrado, J. Mol. Biol. 2007, 366, 436–448.</em>)</li>
<li><strong>flexibility</strong>    (amino acid side chain flexibilitiy scale, <em>[6] R. Bhaskaran, P. K. Ponnuswamy, Int. J. Pept. Protein Res. 1988, 32, 241–255.</em>)</li>
<li><strong>gravy</strong>          (GRAVY hydrophobicity amino acid scale, <em>[7] J. Kyte, R. F. Doolittle, J. Mol. Biol. 1982, 157, 105–132.</em>)</li>
<li><strong>hopp-woods</strong>     (Hopp-Woods amino acid hydrophobicity scale,*[8] T. P. Hopp, K. R. Woods, Proc. Natl. Acad. Sci. 1981, 78, 3824–3828.*)</li>
<li><strong>ISAECI</strong>         (Isotropic Surface Area (ISA) and Electronic Charge Index (ECI) of amino acid side chains, <em>[9] E. R. Collantes, W. J. Dunn, J. Med. Chem. 1995, 38, 2705–2713.</em>)</li>
<li><strong>janin</strong>          (Janin hydrophobicity amino acid scale, [10] J. L. Cornette, K. B. Cease, H. Margalit, J. L. Spouge, J. A. Berzofsky, C. DeLisi, J. Mol. Biol. 1987, 195, 659–685.*)</li>
<li><strong>kytedoolittle</strong>  (Kyte &amp; Doolittle hydrophobicity amino acid scale, <em>[11] J. Kyte, R. F. Doolittle, J. Mol. Biol. 1982, 157, 105–132.</em>)</li>
<li><strong>levitt_alpha</strong>   (Levitt amino acid alpha-helix propensity scale, extracted from <a class="reference external" href="http://web.expasy.org/protscale">http://web.expasy.org/protscale</a>. <em>[12] M. Levitt, Biochemistry 1978, 17, 4277-4285.</em>)</li>
<li><strong>MSS</strong>            (A graph-theoretical index that reflects topological shape and size of amino acid side chains, <em>[13] C. Raychaudhury, A. Banerjee, P. Bag, S. Roy, J. Chem. Inf. Comput. Sci. 1999, 39, 248–254.</em>)</li>
<li><strong>MSW</strong>            (Amino acid scale based on a PCA of the molecular surface based WHIM descriptor (MS-WHIM), extended to natural amino acids, <em>[14] A. Zaliani, E. Gancia, J. Chem. Inf. Comput. Sci 1999, 39, 525–533.</em>)</li>
<li><strong>pepcats</strong>        (modlabs pharmacophoric feature based PEPCATS scale, <em>[15] C. P. Koch, A. M. Perna, M. Pillong, N. K. Todoroff, P. Wrede, G. Folkers, J. A. Hiss, G. Schneider, PLoS Comput. Biol. 2013, 9, e1003088.</em>)</li>
<li><strong>polarity</strong>       (Amino acid polarity scale, <em>[3] J. M. Zimmerman, N. Eliezer, R. Simha, J. Theor. Biol. 1968, 21, 170–201.</em>)</li>
<li><strong>PPCALI</strong>         (modlabs inhouse scale derived from a PCA of 143 amino acid property scales, <em>[15] C. P. Koch, A. M. Perna, M. Pillong, N. K. Todoroff, P. Wrede, G. Folkers, J. A. Hiss, G. Schneider, PLoS Comput. Biol. 2013, 9, e1003088.</em>)</li>
<li><strong>refractivity</strong>   (Relative amino acid refractivity values, <em>[16] T. L. McMeekin, M. Wilensky, M. L. Groves, Biochem. Biophys. Res. Commun. 1962, 7, 151–156.</em>)</li>
<li><strong>t_scale</strong>        (A PCA derived scale based on amino acid side chain properties calculated with 6 different probes of the GRID program, <em>[17] M. Cocchi, E. Johansson, Quant. Struct. Act. Relationships 1993, 12, 1–8.</em>)</li>
<li><strong>TM_tend</strong>        (Amino acid transmembrane propensity scale, extracted from <a class="reference external" href="http://web.expasy.org/protscale">http://web.expasy.org/protscale</a>, <em>[18] Zhao, G., London E. Protein Sci. 2006, 15, 1987-2001.</em>)</li>
<li><strong>z3</strong>             (The original three dimensional Z-scale, <em>[17] S. Hellberg, M. Sjöström, B. Skagerberg, S. Wold, J. Med. Chem. 1987, 30, 1126–1135.</em>)</li>
<li><strong>z5</strong>             (The extended five dimensional Z-scale, <em>[18] M. Sandberg, L. Eriksson, J. Jonsson, M. Sjöström, S. Wold, J. Med. Chem. 1998, 41, 2481–2491.</em>)</li>
</ul>
<p>Further, amino acid scale independent methods can be calculated with help of the <a class="reference internal" href="#modlamp.descriptors.GlobalDescriptor" title="modlamp.descriptors.GlobalDescriptor"><code class="xref py py-class docutils literal"><span class="pre">GlobalDescriptor</span></code></a> class.</p>
<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>seqs</em>, <em>scalename='eisenberg'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.__init__" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seqs</strong> &#8211; a .fasta file with sequences, a list of sequences or a single sequence as string to calculate the
descriptor values for.</li>
<li><strong>scalename</strong> &#8211; {str} name of the amino acid scale (one of the given list above) used to calculate the
descriptor values</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">initialized attributes <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>, <code class="xref py py-attr docutils literal"><span class="pre">names</span></code> and dictionary <code class="xref py py-attr docutils literal"><span class="pre">scale</span></code> with
amino acid scale values of the scale name in <code class="xref py py-attr docutils literal"><span class="pre">scalename</span></code>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span><span class="s1">&#39;pepcats&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KLLKLLKKLLKLLK&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">seqs</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;/Path/to/file.fasta&#39;</span><span class="p">,</span> <span class="s1">&#39;eisenberg&#39;</span><span class="p">)</span>  <span class="c1"># load sequences from .fasta file</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">seqs</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;AFDGHLKI&#39;,&#39;KKLQRSDLLRTK&#39;,&#39;KKLASCNNIPPR&#39;...]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.load_scale">
<code class="descname">load_scale</code><span class="sig-paren">(</span><em>scalename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.load_scale"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.load_scale" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to load amino acid values from a given scale</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>scalename</strong> &#8211; {str} name of the amino acid scale to be loaded.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">loaded amino acid scale values in a dictionary in the attribute <code class="xref py py-attr docutils literal"><span class="pre">scale</span></code>.</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.load_scale" title="modlamp.core.load_scale"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.load_scale()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.read_fasta">
<code class="descname">read_fasta</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.read_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.read_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for loading sequences from a FASTA formatted file into the attributes <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> and
<code class="xref py py-attr docutils literal"><span class="pre">names</span></code>. This method is used by the base class <a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor" title="modlamp.descriptors.PeptideDescriptor"><code class="xref py py-class docutils literal"><span class="pre">PeptideDescriptor</span></code></a> if the input is a FASTA file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> &#8211; {str} .fasta file with sequences and headers to read</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">list of sequences in self.sequences with corresponding sequence names in self.names</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.save_fasta">
<code class="descname">save_fasta</code><span class="sig-paren">(</span><em>outputfile</em>, <em>names=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.save_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.save_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for saving sequences from <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> to a FASTA formatted file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>outputfile</strong> &#8211; {str} filename of the output FASTA file</li>
<li><strong>names</strong> &#8211; {bool} whether sequence names from self.names should be saved as sequence identifiers</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">list of sequences in self.sequences with corresponding sequence names in the attribute <code class="xref py py-attr docutils literal"><span class="pre">names</span></code></p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.calculate_autocorr">
<code class="descname">calculate_autocorr</code><span class="sig-paren">(</span><em>window</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.calculate_autocorr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.calculate_autocorr" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for auto-correlating the amino acid values for a given descriptor scale</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>window</strong> &#8211; {int} correlation window for descriptor calculation in a sliding window approach</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">calculated descriptor numpy.array in self.descriptor</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">,</span><span class="s1">&#39;PPCALI&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">calculate_autocorr</span><span class="p">(</span><span class="mi">7</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[  1.28442339e+00,   1.29025116e+00,   1.03240901e+00, .... ]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(1, 133)</span>
</pre></div>
</div>
<div class="versionchanged">
<p><span class="versionmodified">Changed in version v.2.3.0.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.calculate_crosscorr">
<code class="descname">calculate_crosscorr</code><span class="sig-paren">(</span><em>window</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.calculate_crosscorr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.calculate_crosscorr" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for cross-correlating the amino acid values for a given descriptor scale</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>window</strong> &#8211; {int} correlation window for descriptor calculation in a sliding window approach</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">calculated descriptor numpy.array in self.descriptor</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">,</span><span class="s1">&#39;pepcats&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">calculate_crosscorr</span><span class="p">(</span><span class="mi">7</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.6875    ,  0.46666667,  0.42857143,  0.61538462,  0.58333333, ... ]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(1, 147)</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.calculate_moment">
<code class="descname">calculate_moment</code><span class="sig-paren">(</span><em>window=1000</em>, <em>angle=100</em>, <em>modality='max'</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.calculate_moment"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.calculate_moment" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for calculating the maximum or mean moment of the amino acid values for a given descriptor scale and
window.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>window</strong> &#8211; {int} amino acid window in which to calculate the moment. If the sequence is shorter than the
window, the length of the sequence is taken. So if the default window of 1000 is chosen, for all sequences
shorter than 1000, the <strong>global</strong> hydrophobic moment will be calculated. Otherwise, the maximal
hydrophiobic moment for the chosen window size found in the sequence will be returned.</li>
<li><strong>angle</strong> &#8211; {int} angle in which to calculate the moment. <strong>100</strong> for alpha helices, <strong>180</strong> for beta sheets.</li>
<li><strong>modality</strong> &#8211; {&#8216;max&#8217; or &#8216;mean&#8217;} Calculate respectively maximum or mean hydrophobic moment.</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Calculated descriptor as a numpy.array in self.descriptor and all possible global values in
<code class="xref py py-attr docutils literal"><span class="pre">all_moms</span></code> (needed for the <a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor.calculate_profile" title="modlamp.descriptors.PeptideDescriptor.calculate_profile"><code class="xref py py-func docutils literal"><span class="pre">calculate_profile()</span></code></a> method)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">,</span><span class="s1">&#39;eisenberg&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">calculate_moment</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">modality</span><span class="o">=</span><span class="s1">&#39;max&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.48790226]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.calculate_global">
<code class="descname">calculate_global</code><span class="sig-paren">(</span><em>window=1000</em>, <em>modality='max'</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.calculate_global"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.calculate_global" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for calculating a global / window averaging descriptor value of a given AA scale</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>window</strong> &#8211; {int} amino acid window in which to calculate the moment. If the sequence is shorter than the
window, the length of the sequence is taken.</li>
<li><strong>modality</strong> &#8211; {&#8216;max&#8217; or &#8216;mean&#8217;} Calculate respectively maximum or mean hydrophobic moment.</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Calculated descriptor as numpy.array in self.descriptor and all possible global values in
<code class="xref py py-attr docutils literal"><span class="pre">all_globs</span></code> (needed for the <a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor.calculate_profile" title="modlamp.descriptors.PeptideDescriptor.calculate_profile"><code class="xref py py-func docutils literal"><span class="pre">calculate_profile()</span></code></a> method)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">,</span><span class="s1">&#39;eisenberg&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">calculate_global</span><span class="p">(</span><span class="n">window</span><span class="o">=</span><span class="mi">1000</span><span class="p">,</span> <span class="n">modality</span><span class="o">=</span><span class="s1">&#39;max&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.44875]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.calculate_profile">
<code class="descname">calculate_profile</code><span class="sig-paren">(</span><em>prof_type='uH'</em>, <em>window=7</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.calculate_profile"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.calculate_profile" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for calculating hydrophobicity or hydrophobic moment profiles for given sequences and fitting for
slope and intercept. The hydrophobicity scale used is &#8220;eisenberg&#8221;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>prof_type</strong> &#8211; prof_type of profile, available: &#8216;H&#8217; for hydrophobicity or &#8216;uH&#8217; for hydrophobic moment</li>
<li><strong>window</strong> &#8211; {int} size of sliding window used (odd-numbered).</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">Fitted slope and intercept of calculated profile for every given sequence in self.descriptor</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;KLLKLLKKVVGALG&#39;</span><span class="p">,</span><span class="s1">&#39;kytedoolittle&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">calculate_profile</span><span class="p">(</span><span class="n">prof_type</span><span class="o">=</span><span class="s1">&#39;H&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.03731293,  0.19246599]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.count_aa">
<code class="descname">count_aa</code><span class="sig-paren">(</span><em>scale='relative'</em>, <em>append=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.count_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.count_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for producing the amino acid distribution for the given sequences as a descriptor</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>scale</strong> &#8211; {&#8216;absolute&#8217; or &#8216;relative&#8217;} defines whether counts or frequencies are given for each AA</li>
<li><strong>append</strong> &#8211; {boolean} whether the produced descriptor values should be appended to the existing ones in the
attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">the amino acid distributions for every sequence individually in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="s1">&#39;ACDEFGHIKLMNPQRSTVWY&#39;</span><span class="p">)</span> <span class="c1"># aa_count() does not depend on the descriptor scale</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">count_aa</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[ 0.05,  0.05,  0.05,  0.05,  0.05,  0.05,  0.05,  0.05,  0.05, ... ]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">AMP</span><span class="o">.</span><span class="n">descriptor</span><span class="o">.</span><span class="n">shape</span>
<span class="go">(1, 20)</span>
</pre></div>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.count_aa" title="modlamp.core.count_aa"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.count_aa()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.feature_scaling">
<code class="descname">feature_scaling</code><span class="sig-paren">(</span><em>stype='standard'</em>, <em>fit=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.feature_scaling"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.feature_scaling" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for feature scaling of the calculated descriptor matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>stype</strong> &#8211; {&#8216;standard&#8217; or &#8216;minmax&#8217;} type of scaling to be used</li>
<li><strong>fit</strong> &#8211; {boolean} defines whether the used scaler is first fitting on the data (True) or
whether the already fitted scaler in <code class="xref py py-attr docutils literal"><span class="pre">scaler</span></code> should be used to transform (False).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">scaled descriptor values in <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[0.155],[0.34],[0.16235294],[-0.08842105],[0.116]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">feature_scaling</span><span class="p">(</span><span class="nb">type</span><span class="o">=</span><span class="s1">&#39;minmax&#39;</span><span class="p">,</span><span class="n">fit</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">array([[0.56818182],[1.],[0.5853447],[0.],[0.47714988]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.feature_shuffle">
<code class="descname">feature_shuffle</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.feature_shuffle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.feature_shuffle" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for shuffling feature columns randomly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">descriptor matrix with shuffled feature columns in <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">descriptor</span>
<span class="go">array([[0.80685625,167.05234375,39.56818125,-0.26338667,155.16888667,33.48778]])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">feature_shuffle</span><span class="p">()</span>
<span class="go">array([[155.16888667,-0.26338667,167.05234375,0.80685625,39.56818125,33.48778]])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.sequence_order_shuffle">
<code class="descname">sequence_order_shuffle</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.sequence_order_shuffle"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.sequence_order_shuffle" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for shuffling sequence order in self.sequences.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">sequences in <code class="xref py py-attr docutils literal"><span class="pre">self.sequences</span></code> with shuffled order in the list.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;LILRALKGAARALKVA&#39;,&#39;VKIAKIALKIIKGLG&#39;,&#39;VGVRLIKGIGRVARGAI&#39;,&#39;LRGLRGVIRGGKAIVRVGK&#39;,&#39;GGKLVRLIARIGKGV&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">sequence_order_shuffle</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">D</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;VGVRLIKGIGRVARGAI&#39;,&#39;LILRALKGAARALKVA&#39;,&#39;LRGLRGVIRGGKAIVRVGK&#39;,&#39;GGKLVRLIARIGKGV&#39;,&#39;VKIAKIALKIIKGLG&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.filter_unnatural">
<code class="descname">filter_unnatural</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.filter_unnatural"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.filter_unnatural" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences with unnatural amino acids from <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
:return: Filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></p>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_unnatural" title="modlamp.core.filter_unnatural"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_unnatural()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.filter_values">
<code class="descname">filter_values</code><span class="sig-paren">(</span><em>values</em>, <em>operator='=='</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.filter_values"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.filter_values" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter the descriptor matrix in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code> for a given list of value (same
size as the number of features in the descriptor matrix!)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>values</strong> &#8211; List of values to filter</li>
<li><strong>operator</strong> &#8211; filter criterion, available are all SQL like operators: <code class="docutils literal"><span class="pre">==</span></code>, <code class="docutils literal"><span class="pre">&lt;</span></code>, <code class="docutils literal"><span class="pre">&gt;</span></code>, <code class="docutils literal"><span class="pre">&lt;=``and</span> <span class="pre">``&gt;=</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">filtered descriptor matrix and updated sequences in the corresponding attributes.</p>
</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_values" title="modlamp.core.filter_values"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_values()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.filter_duplicates">
<code class="descname">filter_duplicates</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.filter_duplicates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.filter_duplicates" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter duplicates in the sequences from the class attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequences list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_sequences" title="modlamp.core.filter_sequences"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_sequences()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.5.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.keep_natural_aa">
<code class="descname">keep_natural_aa</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.keep_natural_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.keep_natural_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences that do not contain natural amino acids. If the sequence contains a character
that is not in [&#8216;A&#8217;,&#8217;C&#8217;,&#8217;D,&#8217;E&#8217;,&#8217;F&#8217;,&#8217;G&#8217;,&#8217;H&#8217;,&#8217;I&#8217;,&#8217;K&#8217;,&#8217;L&#8217;,&#8217;M&#8217;,&#8217;N&#8217;,&#8217;P&#8217;,&#8217;Q&#8217;,&#8217;R&#8217;,&#8217;S&#8217;,&#8217;T&#8217;,&#8217;V&#8217;,&#8217;W&#8217;,&#8217;Y&#8217;].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>. The other attributes are also filtered
accordingly.</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.keep_natural_aa" title="modlamp.core.keep_natural_aa"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.keep_natural_aa()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.5.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.filter_aa">
<code class="descname">filter_aa</code><span class="sig-paren">(</span><em>aminoacids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.filter_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.filter_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter sequences and corresponding descriptor values, if the sequences contain any of the given
amino acids in the argument list <strong>aminoacids</strong>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aminoacids</strong> &#8211; List/array of amino acids to filter for</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">filtered descriptor matrix and updated sequences and names in the corresponding attributes.</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_aa" title="modlamp.core.filter_aa"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_aa()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.filter_sequences">
<code class="descname">filter_sequences</code><span class="sig-paren">(</span><em>sequences</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.filter_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.filter_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out entries for given sequences in <em>sequences</em> out of a descriptor instance. All
corresponding fields of these sequences (<em>descriptor</em>, <em>name</em>) are deleted as well. The method returns an
updated descriptor instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>sequences</strong> &#8211; list of sequences to be filtered out of the whole instance, including corresponding data</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">updated instance</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.filter_sequences" title="modlamp.core.filter_sequences"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.filter_sequences()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.4.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.random_selection">
<code class="descname">random_selection</code><span class="sig-paren">(</span><em>num</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.random_selection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.random_selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to randomly select a specified number of sequences (with names and descriptors if present) out of a
given descriptor instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>num</strong> &#8211; {int} number of entries to be randomly selected</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">updated instance</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.random_selection" title="modlamp.core.random_selection"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.random_selection()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.3.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.minmax_selection">
<code class="descname">minmax_selection</code><span class="sig-paren">(</span><em>iterations</em>, <em>distmetric='euclidean'</em>, <em>randseed=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.minmax_selection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.minmax_selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to select a specified number of sequences out of a given descriptor instance according to the
minmax algorithm.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>iterations</strong> &#8211; {int} Number of sequences to retrieve.</li>
<li><strong>distmetric</strong> &#8211; Distance metric to calculate the distances between the sequences in descriptor space.
Choose from scipy.spacial.distance (<a class="reference external" href="http://docs.scipy.org/doc/scipy/reference/spatial.distance.html">http://docs.scipy.org/doc/scipy/reference/spatial.distance.html</a>).
E.g. &#8216;euclidean&#8217;, &#8216;minkowsky&#8217;.</li>
<li><strong>randseed</strong> &#8211; {int} Set a random seed for numpy to pick the first sequence.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">updated instance</p>
</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.minmax_selection" title="modlamp.core.minmax_selection"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.minmax_selection()</span></code></a></p>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.6.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.load_descriptordata">
<code class="descname">load_descriptordata</code><span class="sig-paren">(</span><em>filename</em>, <em>delimiter='</em>, <em>'</em>, <em>targets=False</em>, <em>header=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.load_descriptordata"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.load_descriptordata" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to load any data file with sequences and descriptor values and save it to a new insatnce of the
class <a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor" title="modlamp.descriptors.PeptideDescriptor"><code class="xref py py-class docutils literal"><span class="pre">modlamp.descriptors.PeptideDescriptor</span></code></a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The data file should <strong>not</strong> have any headers</p>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> &#8211; filenam of the data file to be loaded</li>
<li><strong>delimiter</strong> &#8211; column delimiter</li>
<li><strong>targets</strong> &#8211; {boolean} whether last column in the file contains a target class vector</li>
<li><strong>header</strong> &#8211; {int} number of header lines to skip in the file</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">loaded sequences, descriptor values and targets in the corresponding attributes.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.descriptors.PeptideDescriptor.save_descriptor">
<code class="descname">save_descriptor</code><span class="sig-paren">(</span><em>filename</em>, <em>delimiter='</em>, <em>'</em>, <em>targets=None</em>, <em>header=''</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/descriptors.html#PeptideDescriptor.save_descriptor"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.descriptors.PeptideDescriptor.save_descriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to save the descriptor values to a .csv/.txt file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> &#8211; filename of the output file</li>
<li><strong>delimiter</strong> &#8211; column delimiter</li>
<li><strong>targets</strong> &#8211; target class vector to be added to descriptor (same length as <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>)</li>
<li><strong>header</strong> &#8211; {str} header to be written at the beginning of the file</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">output file with peptide names and descriptor values</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-modlamp.sequences">
<span id="modlamp-sequences"></span><h2>2.2. modlamp.sequences<a class="headerlink" href="#module-modlamp.sequences" title="Permalink to this headline">¶</a></h2>
<p>This module incorporates different classes to generate peptide sequences with different characteristics from scratch.
The following classes are available:</p>
<table border="1" class="docutils">
<colgroup>
<col width="26%" />
<col width="74%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Class</th>
<th class="head">Characteristics</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#modlamp.sequences.Random" title="modlamp.sequences.Random"><code class="xref py py-class docutils literal"><span class="pre">Random</span></code></a></td>
<td>Generates random sequences with a specified amino acid distribution.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.sequences.Helices" title="modlamp.sequences.Helices"><code class="xref py py-class docutils literal"><span class="pre">Helices</span></code></a></td>
<td>Generates presumed amphipathic helical sequences with a hydrophobic moment.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.sequences.Kinked" title="modlamp.sequences.Kinked"><code class="xref py py-class docutils literal"><span class="pre">Kinked</span></code></a></td>
<td>Generates presumed amphipathic helices with a kink (Pro residue).</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.sequences.Oblique" title="modlamp.sequences.Oblique"><code class="xref py py-class docutils literal"><span class="pre">Oblique</span></code></a></td>
<td>Generates presumed oblique oriented sequences in presence of libid membranes.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.sequences.Centrosymmetric" title="modlamp.sequences.Centrosymmetric"><code class="xref py py-class docutils literal"><span class="pre">Centrosymmetric</span></code></a></td>
<td>Generates centrosymmetric sequences with a symmetry axis.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.sequences.MixedLibrary" title="modlamp.sequences.MixedLibrary"><code class="xref py py-class docutils literal"><span class="pre">MixedLibrary</span></code></a></td>
<td>Generates a mixed library of sequences of all other classes.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.sequences.Hepahelices" title="modlamp.sequences.Hepahelices"><code class="xref py py-class docutils literal"><span class="pre">Hepahelices</span></code></a></td>
<td>Generates presumed amphipathic helices with a heparin-binding-domain.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.sequences.AMPngrams" title="modlamp.sequences.AMPngrams"><code class="xref py py-class docutils literal"><span class="pre">AMPngrams</span></code></a></td>
<td>Generates sequences from most frequent ngrams in the APD3.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.sequences.HelicesACP" title="modlamp.sequences.HelicesACP"><code class="xref py py-class docutils literal"><span class="pre">HelicesACP</span></code></a></td>
<td>Generates sequences with the amino acid probabiliy of helical ACPs.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.sequences.AmphipathicArc" title="modlamp.sequences.AmphipathicArc"><code class="xref py py-class docutils literal"><span class="pre">AmphipathicArc</span></code></a></td>
<td>Generates presumed amphipathic helices with controlled hydrophobic arc size.</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">During the process of sequence generation, duplicates are only removed for the <a class="reference internal" href="#modlamp.sequences.MixedLibrary" title="modlamp.sequences.MixedLibrary"><code class="xref py py-class docutils literal"><span class="pre">MixedLibrary</span></code></a>
class. To remove duplicates, call the class methods <code class="xref py py-func docutils literal"><span class="pre">self.filter_duplicates()</span></code>.</p>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.BaseSequence" title="modlamp.core.BaseSequence"><code class="xref py py-class docutils literal"><span class="pre">modlamp.core.BaseSequence</span></code></a> from which all classes in this module inherit.</p>
</div>
<dl class="class">
<dt id="modlamp.sequences.Random">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">Random</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Random"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Random" title="Permalink to this definition">¶</a></dt>
<dd><p>Class for random peptide sequences
This class incorporates methods for generating peptide random peptide sequences of defined length.
The amino acid probabilities can be chosen from different probabilities:</p>
<ul class="simple">
<li><strong>rand</strong>: equal probabilities for all amino acids</li>
<li><strong>AMP</strong>: amino acid probabilities taken from the antimicrobial peptide database <a class="reference external" href="http://aps.unmc.edu/AP/statistic/statistic.php">APD3</a>, March 17, 2016, containing 2674 sequences.</li>
<li><strong>AMPnoCM</strong>: same amino acid probabilities as <strong>AMP</strong> but lacking Cys and Met (for synthesizability)</li>
<li><strong>randnoCM</strong>: equal probabilities for all amino acids, except 0.0 for both Cys and Met (for synthesizability)</li>
</ul>
<p>The probability values for all natural AA can be found in the following table:</p>
<table border="1" class="docutils">
<colgroup>
<col width="9%" />
<col width="13%" />
<col width="19%" />
<col width="28%" />
<col width="31%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">AA</th>
<th class="head">rand</th>
<th class="head">AMP</th>
<th class="head">AMPnoCM</th>
<th class="head">randnoCM</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>A</td>
<td>0.05</td>
<td>0.0766</td>
<td>0.0812275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>C</td>
<td>0.05</td>
<td>0.071</td>
<td>0.0</td>
<td>0.0</td>
</tr>
<tr class="row-even"><td>D</td>
<td>0.05</td>
<td>0.026</td>
<td>0.0306275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>E</td>
<td>0.05</td>
<td>0.0264</td>
<td>0.0310275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>F</td>
<td>0.05</td>
<td>0.0405</td>
<td>0.0451275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>G</td>
<td>0.05</td>
<td>0.1172</td>
<td>0.1218275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>H</td>
<td>0.05</td>
<td>0.021</td>
<td>0.0256275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>I</td>
<td>0.05</td>
<td>0.061</td>
<td>0.0656275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>K</td>
<td>0.05</td>
<td>0.0958</td>
<td>0.1004275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>L</td>
<td>0.05</td>
<td>0.0838</td>
<td>0.0884275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>M</td>
<td>0.05</td>
<td>0.0123</td>
<td>0.0</td>
<td>0.0</td>
</tr>
<tr class="row-odd"><td>N</td>
<td>0.05</td>
<td>0.0386</td>
<td>0.0432275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>P</td>
<td>0.05</td>
<td>0.0463</td>
<td>0.0509275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>Q</td>
<td>0.05</td>
<td>0.0251</td>
<td>0.0297275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>R</td>
<td>0.05</td>
<td>0.0545</td>
<td>0.0591275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>S</td>
<td>0.05</td>
<td>0.0613</td>
<td>0.0659275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>T</td>
<td>0.05</td>
<td>0.0455</td>
<td>0.0501275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>V</td>
<td>0.05</td>
<td>0.0572</td>
<td>0.0618275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>W</td>
<td>0.05</td>
<td>0.0155</td>
<td>0.0201275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>Y</td>
<td>0.05</td>
<td>0.0244</td>
<td>0.0290275</td>
<td>0.05555555</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="modlamp.sequences.Random.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><em>proba='rand'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Random.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Random.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to actually generate the sequences.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>proba</strong> &#8211; AA probability to be used to generate sequences. Available: AMP, AMPnoCM, rand, randnoCM</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">A list of random AMP sequences with defined AA probabilities</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">Random</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">20</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">(</span><span class="n">proba</span><span class="o">=</span><span class="s1">&#39;AMP&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;CYGALWHIFV&#39;,&#39;NIVRHHAPSTVIK&#39;,&#39;LCPNPILGIV&#39;,&#39;TAVVRGKESLTP&#39;,&#39;GTGSVCKNSCRGRFGIIAF&#39;,&#39;VIIGPSYGDAEYA&#39;]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.Helices">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">Helices</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Helices"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Helices" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for peptide sequences probable to form helices.</p>
<p>This class incorporates methods for generating presumed amphipathic alpha-helical peptide sequences.
These sequences are generated by placing basic residues along the sequence with distance 3-4 AA to each other.
The remaining empty spots are filled up by hydrophobic AAs.</p>
<dl class="method">
<dt id="modlamp.sequences.Helices.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Helices.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Helices.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate amphipathic helical sequences with class features defined in <a class="reference internal" href="#modlamp.sequences.Helices" title="modlamp.sequences.Helices"><code class="xref py py-class docutils literal"><span class="pre">Helices()</span></code></a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">In the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>: a list of sequences with presumed amphipathic helical structure.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">h</span> <span class="o">=</span> <span class="n">Helices</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">21</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KGIKVILKLAKAGVKAVRL&#39;,&#39;IILKVGKV&#39;,&#39;IAKAGRAIIK&#39;,&#39;LKILKVVGKGIRLIVRIIKAL&#39;,&#39;KAGKLVAKGAKVAAKAIKI&#39;]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.Kinked">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">Kinked</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Kinked"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Kinked" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for peptide sequences probable to form helices with a kink.
This class incorporates methods for presumed kinked amphipathic alpha-helical peptide sequences:
Sequences are generated by placing basic residues along the sequence with distance 3-4 AA to each other.
The remaining spots are filled up by hydrophobic AAs. Then, a basic residue is replaced by proline, presumably
leading to a kink in the hydrophobic face of the amphipathic helices.</p>
<dl class="method">
<dt id="modlamp.sequences.Kinked.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Kinked.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Kinked.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to actually generate the presumed kinked sequences with features defined in the class instances.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">sequence list with strings stored in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">k</span> <span class="o">=</span> <span class="n">Kinked</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">28</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">k</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">k</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;IILRLHPIG&#39;,&#39;ARGAKVAIKAIRGIAPGGRVVAKVVKVG&#39;,&#39;GGKVGRGVAFLVRIILK&#39;,&#39;KAVKALAKGAPVILCVAKVI&#39;, ...]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.Oblique">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">Oblique</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Oblique"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Oblique" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for oblique sequences with a so called linear hydrophobicity gradient.</p>
<p>This class incorporates methods for generating peptide sequences with a linear hydrophobicity gradient, meaning that
these sequences have a hydrophobic tail. This feature gives rise to the hypothesis that they orient themselves
tilted/oblique in membrane environment.</p>
<dl class="method">
<dt id="modlamp.sequences.Oblique.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Oblique.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Oblique.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate the possible oblique sequences.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">A list of sequences in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">o</span> <span class="o">=</span> <span class="n">Oblique</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">30</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">o</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">o</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;GLLKVIRIAAKVLKVAVLVGIIAI&#39;,&#39;AIGKAGRLALKVIKVVIKVALILLAAVA&#39;,&#39;KILRAAARVIKGGIKAIVIL&#39;,&#39;VRLVKAIGKLLRIILRLARLAVGGILA&#39;]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.Centrosymmetric">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">Centrosymmetric</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Centrosymmetric"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Centrosymmetric" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for peptide sequences produced out of 7 AA centro-symmetric blocks yielding peptides of length
14 or 21 AA (2*7 or 3*7).</p>
<p>This class incorporates methods to generate special peptide sequences with an overall presumed
hydrophobic moment. Sequences are generated by centro-symmetric blocks of seven amino acids. Two or three blocks
are added to build a final sequence of length 14 or 21 amino acids length. If the method <a class="reference internal" href="#modlamp.sequences.Centrosymmetric.generate_symmetric" title="modlamp.sequences.Centrosymmetric.generate_symmetric"><code class="xref py py-func docutils literal"><span class="pre">generate_symmetric()</span></code></a>
is used, two or three identical blocks are concatenated. If the method <a class="reference internal" href="#modlamp.sequences.Centrosymmetric.generate_asymmetric" title="modlamp.sequences.Centrosymmetric.generate_asymmetric"><code class="xref py py-func docutils literal"><span class="pre">generate_asymmetric()</span></code></a> is used,
two or three different blocks are concatenated.</p>
<dl class="method">
<dt id="modlamp.sequences.Centrosymmetric.generate_symmetric">
<code class="descname">generate_symmetric</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Centrosymmetric.generate_symmetric"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Centrosymmetric.generate_symmetric" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="#modlamp.sequences.Centrosymmetric.generate_symmetric" title="modlamp.sequences.Centrosymmetric.generate_symmetric"><code class="xref py py-func docutils literal"><span class="pre">generate_symmetric()</span></code></a> method generates overall symmetric sequences out of two or three blocks of
identical centro-symmetric sequence blocks of 7 amino acids. The resulting sequence presumably has a large
hydrophobic moment.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">In the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>: centro-symmetric peptide sequences of the form [h,+,h,a,h,+,h] with
h = hydrophobic AA, + = basic AA, a = anchor AA (F,Y,W,(P)), sequence length is 14 or 21 AA</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">s</span> <span class="o">=</span> <span class="n">Centrosymmetric</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">s</span><span class="o">.</span><span class="n">generate_symmetric</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">s</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;ARIFIRAARIFIRA&#39;,&#39;GRIYIRGGRIYIRGGRIYIRG&#39;,&#39;IRGFGRIIRGFGRIIRGFGRI&#39;,&#39;GKAYAKGGKAYAKG&#39;,&#39;AKGYGKAAKGYGKAAKGYGKA&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.sequences.Centrosymmetric.generate_asymmetric">
<code class="descname">generate_asymmetric</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Centrosymmetric.generate_asymmetric"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Centrosymmetric.generate_asymmetric" title="Permalink to this definition">¶</a></dt>
<dd><p>The <a class="reference internal" href="#modlamp.sequences.Centrosymmetric.generate_asymmetric" title="modlamp.sequences.Centrosymmetric.generate_asymmetric"><code class="xref py py-func docutils literal"><span class="pre">generate_asymmetric()</span></code></a> method generates overall asymmetric sequences out of two or three blocks of
different centro-symmetric sequence blocks of 7 amino acids. The resulting sequence presumably has a large
hydrophobic    moment.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">In the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>: a list of peptide sequences as strings of length 14 or 21</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">S</span> <span class="o">=</span> <span class="n">Centrosymmetric</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">S</span><span class="o">.</span><span class="n">generate_asymmetric</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">S</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;GRLFLRGAKGFGKAVRVWVRV&#39;,&#39;IKGWGKILKLYLKL&#39;,&#39;LKAYAKLVKAWAKVLRLFLRL&#39;,&#39;IRLWLRIIKAFAKI&#39;,&#39;LRIFIRLVKLWLKVIRLWLRI&#39;]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.AmphipathicArc">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">AmphipathicArc</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AmphipathicArc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AmphipathicArc" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for generating positively-charged amphipathic peptide sequences based on an alpha-helix pattern with
different arc sizes.</p>
<p>The probability values for the Hydrophobic and Polar positions of the helix can be found in the following table:</p>
<table border="1" class="docutils">
<colgroup>
<col width="25%" />
<col width="33%" />
<col width="42%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">AA</th>
<th class="head">Hydr</th>
<th class="head">Polar</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>A</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-odd"><td>C</td>
<td>0.00</td>
<td>0.00</td>
</tr>
<tr class="row-even"><td>D</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-odd"><td>E</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-even"><td>F</td>
<td>0.20</td>
<td>0.00</td>
</tr>
<tr class="row-odd"><td>G</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-even"><td>H</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-odd"><td>I</td>
<td>0.20</td>
<td>0.00</td>
</tr>
<tr class="row-even"><td>K</td>
<td>0.00</td>
<td>0.25</td>
</tr>
<tr class="row-odd"><td>L</td>
<td>0.20</td>
<td>0.00</td>
</tr>
<tr class="row-even"><td>M</td>
<td>0.00</td>
<td>0.00</td>
</tr>
<tr class="row-odd"><td>N</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-even"><td>P</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-odd"><td>Q</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-even"><td>R</td>
<td>0.00</td>
<td>0.25</td>
</tr>
<tr class="row-odd"><td>S</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-even"><td>T</td>
<td>0.00</td>
<td>0.05</td>
</tr>
<tr class="row-odd"><td>V</td>
<td>0.20</td>
<td>0.00</td>
</tr>
<tr class="row-even"><td>W</td>
<td>0.20</td>
<td>0.00</td>
</tr>
<tr class="row-odd"><td>Y</td>
<td>0.00</td>
<td>0.05</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="modlamp.sequences.AmphipathicArc.generate_arc">
<code class="descname">generate_arc</code><span class="sig-paren">(</span><em>arcsize=160</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AmphipathicArc.generate_arc"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AmphipathicArc.generate_arc" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate the possible amphipathic helices with defined hydrophobic arc sizes.
:param arcsize: {int} to choose among 80, 120, 160, 200, 240.
:return: A list of sequences in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
:Example:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_hel</span> <span class="o">=</span> <span class="n">AmphipathicArc</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="mi">25</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_hel</span><span class="o">.</span><span class="n">generate_arc</span><span class="p">(</span><span class="mi">80</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_hel</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;YLYANLRQE&#39;, &#39;GVKPRIK&#39;, &#39;RWKKKVKDSVKDFEKRFKDIEKRIQRKLA&#39;, &#39;KIKEQLRNSVSGWHRN&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.sequences.AmphipathicArc.generate_mixed_arcs">
<code class="descname">generate_mixed_arcs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AmphipathicArc.generate_mixed_arcs"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AmphipathicArc.generate_mixed_arcs" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate sequences of mixed arc sizes.
:return: A list of sequences in <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>
:Example:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_hel</span> <span class="o">=</span> <span class="n">Amphipathic_arc</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">30</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_hel</span><span class="o">.</span><span class="n">generate_mixed_arcs</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_hel</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KIRRAFRNNLK&#39;, &#39;PFWKRWARWFKRWHRKLKTVFAKVTALL&#39;, &#39;AVFRTIKAVF&#39;, &#39;AIVHFFLTFLAELWQFLK&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.sequences.AmphipathicArc.make_H_gradient">
<code class="descname">make_H_gradient</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AmphipathicArc.make_H_gradient"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AmphipathicArc.make_H_gradient" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to mutate the generated sequences to have a hydrophobic gradient by substituting the last third of
the sequence amino acids to  hydrophobic.
:return: A list of sequences in <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>
:Example:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_grad</span> <span class="o">=</span> <span class="n">AmphipathicArc</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">30</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_grad</span><span class="o">.</span><span class="n">generate_mixed_arcs</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_grad</span><span class="o">.</span><span class="n">make_H_gradient</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">amphi_grad</span><span class="o">.</span><span class="n">sequences</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.HelicesACP">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">HelicesACP</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#HelicesACP"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.HelicesACP" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for peptides sequences with the amino acid probability of alpha-helical ACPs.</p>
<p>This class incorporates methods for generating presumed alpha-helical peptides with the amino acid probability
distribution of alpha-helical ACPs. For each of the positions in the helix (1-18) the amino acid distribution among
62 anuran and hymenopteran alpha-helical ACPs was computed and is used to design the new sequences
(Gabernet et al., MedChemComm 2016).</p>
<table border="1" class="docutils">
<colgroup>
<col width="4%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">AA</th>
<th class="head">pos0</th>
<th class="head">pos1</th>
<th class="head">pos2</th>
<th class="head">pos3</th>
<th class="head">pos4</th>
<th class="head">pos5</th>
<th class="head">pos6</th>
<th class="head">pos7</th>
<th class="head">pos8</th>
<th class="head">pos9</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>A</td>
<td>0.048</td>
<td>0.000</td>
<td>0.000</td>
<td>0.048</td>
<td>0.016</td>
<td>0.129</td>
<td>0.032</td>
<td>0.097</td>
<td>0.193</td>
<td>0.500</td>
</tr>
<tr class="row-odd"><td>C</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>D</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.274</td>
<td>0.017</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>E</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.064</td>
<td>0.000</td>
<td>0.016</td>
</tr>
<tr class="row-even"><td>F</td>
<td>0.161</td>
<td>0.048</td>
<td>0.306</td>
<td>0.000</td>
<td>0.048</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
</tr>
<tr class="row-odd"><td>G</td>
<td>0.645</td>
<td>0.000</td>
<td>0.177</td>
<td>0.145</td>
<td>0.000</td>
<td>0.016</td>
<td>0.258</td>
<td>0.113</td>
<td>0.064</td>
<td>0.081</td>
</tr>
<tr class="row-even"><td>H</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.032</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>I</td>
<td>0.048</td>
<td>0.113</td>
<td>0.016</td>
<td>0.081</td>
<td>0.338</td>
<td>0.274</td>
<td>0.000</td>
<td>0.048</td>
<td>0.145</td>
<td>0.064</td>
</tr>
<tr class="row-even"><td>K</td>
<td>0.000</td>
<td>0.032</td>
<td>0.016</td>
<td>0.129</td>
<td>0.129</td>
<td>0.000</td>
<td>0.387</td>
<td>0.338</td>
<td>0.048</td>
<td>0.032</td>
</tr>
<tr class="row-odd"><td>L</td>
<td>0.048</td>
<td>0.710</td>
<td>0.129</td>
<td>0.048</td>
<td>0.096</td>
<td>0.322</td>
<td>0.210</td>
<td>0.064</td>
<td>0.113</td>
<td>0.064</td>
</tr>
<tr class="row-even"><td>M</td>
<td>0.000</td>
<td>0.016</td>
<td>0.016</td>
<td>0.048</td>
<td>0.016</td>
<td>0.032</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>N</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>P</td>
<td>0.000</td>
<td>0.000</td>
<td>0.145</td>
<td>0.016</td>
<td>0.032</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>Q</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>R</td>
<td>0.000</td>
<td>0.016</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.016</td>
</tr>
<tr class="row-odd"><td>S</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.129</td>
<td>0.032</td>
<td>0.032</td>
<td>0.048</td>
<td>0.177</td>
<td>0.000</td>
<td>0.032</td>
</tr>
<tr class="row-even"><td>T</td>
<td>0.016</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.048</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.016</td>
</tr>
<tr class="row-odd"><td>V</td>
<td>0.016</td>
<td>0.016</td>
<td>0.016</td>
<td>0.016</td>
<td>0.209</td>
<td>0.161</td>
<td>0.016</td>
<td>0.048</td>
<td>0.338</td>
<td>0.161</td>
</tr>
<tr class="row-even"><td>W</td>
<td>0.000</td>
<td>0.000</td>
<td>0.129</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>Y</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils">
<colgroup>
<col width="5%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
<col width="12%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">AA</th>
<th class="head">pos10</th>
<th class="head">pos11</th>
<th class="head">pos12</th>
<th class="head">pos13</th>
<th class="head">pos14</th>
<th class="head">pos15</th>
<th class="head">pos16</th>
<th class="head">pos17</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>A</td>
<td>0.048</td>
<td>0.113</td>
<td>0.100</td>
<td>0.185</td>
<td>0.078</td>
<td>0.120</td>
<td>0.170</td>
<td>0.167</td>
</tr>
<tr class="row-odd"><td>C</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.024</td>
<td>0.194</td>
</tr>
<tr class="row-even"><td>D</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>E</td>
<td>0.048</td>
<td>0.016</td>
<td>0.000</td>
<td>0.019</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>F</td>
<td>0.000</td>
<td>0.097</td>
<td>0.067</td>
<td>0.019</td>
<td>0.000</td>
<td>0.020</td>
<td>0.146</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>G</td>
<td>0.225</td>
<td>0.032</td>
<td>0.067</td>
<td>0.204</td>
<td>0.137</td>
<td>0.100</td>
<td>0.000</td>
<td>0.055</td>
</tr>
<tr class="row-even"><td>H</td>
<td>0.000</td>
<td>0.210</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.160</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>I</td>
<td>0.032</td>
<td>0.065</td>
<td>0.183</td>
<td>0.000</td>
<td>0.000</td>
<td>0.100</td>
<td>0.268</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>K</td>
<td>0.419</td>
<td>0.081</td>
<td>0.000</td>
<td>0.037</td>
<td>0.294</td>
<td>0.040</td>
<td>0.024</td>
<td>0.028</td>
</tr>
<tr class="row-odd"><td>L</td>
<td>0.032</td>
<td>0.032</td>
<td>0.283</td>
<td>0.240</td>
<td>0.039</td>
<td>0.280</td>
<td>0.073</td>
<td>0.222</td>
</tr>
<tr class="row-even"><td>M</td>
<td>0.000</td>
<td>0.000</td>
<td>0.033</td>
<td>0.000</td>
<td>0.020</td>
<td>0.020</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>N</td>
<td>0.032</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.020</td>
<td>0.020</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>P</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.130</td>
<td>0.176</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>Q</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.019</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
<tr class="row-even"><td>R</td>
<td>0.016</td>
<td>0.016</td>
<td>0.000</td>
<td>0.019</td>
<td>0.019</td>
<td>0.000</td>
<td>0.049</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>S</td>
<td>0.096</td>
<td>0.048</td>
<td>0.017</td>
<td>0.000</td>
<td>0.157</td>
<td>0.100</td>
<td>0.000</td>
<td>0.056</td>
</tr>
<tr class="row-even"><td>T</td>
<td>0.016</td>
<td>0.113</td>
<td>0.000</td>
<td>0.019</td>
<td>0.039</td>
<td>0.020</td>
<td>0.000</td>
<td>0.056</td>
</tr>
<tr class="row-odd"><td>V</td>
<td>0.000</td>
<td>0.145</td>
<td>0.250</td>
<td>0.111</td>
<td>0.020</td>
<td>0.020</td>
<td>0.220</td>
<td>0.222</td>
</tr>
<tr class="row-even"><td>W</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.024</td>
<td>0.000</td>
</tr>
<tr class="row-odd"><td>Y</td>
<td>0.000</td>
<td>0.016</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
<td>0.000</td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="modlamp.sequences.HelicesACP.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#HelicesACP.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.HelicesACP.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate the sequences with the mentioned amino acid probabilities.
:return: A list of potentially helical peptides with the amino acid distribution of ACP helical peptides
according to the position in the helix wheel.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">helACP</span> <span class="o">=</span> <span class="n">Helix_ACP</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">18</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">helACP</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">helACP</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;FLFDVAKKVAGTALT&#39;, &#39;GLGIILGAGG&#39;, &#39;GLRIKLGVWAKKA&#39;, &#39;GFWGFIKTI&#39;]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.MixedLibrary">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">MixedLibrary</code><span class="sig-paren">(</span><em>seqnum</em>, <em>centrosymmetric=1</em>, <em>centroasymmetric=1</em>, <em>helix=1</em>, <em>kinked=1</em>, <em>oblique=1</em>, <em>rand=1</em>, <em>randAMP=1</em>, <em>randAMPnoCM=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#MixedLibrary"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.MixedLibrary" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for holding a virtual peptide library.</p>
<p>This class <a class="reference internal" href="#modlamp.sequences.MixedLibrary" title="modlamp.sequences.MixedLibrary"><code class="xref py py-class docutils literal"><span class="pre">MixedLibrary</span></code></a> incorporates methods to generate a virtual peptide library composed out of different
sub-libraries. The available library subtypes are all from the classes <a class="reference internal" href="#modlamp.sequences.Centrosymmetric" title="modlamp.sequences.Centrosymmetric"><code class="xref py py-class docutils literal"><span class="pre">Centrosymmetric</span></code></a>, <a class="reference internal" href="#modlamp.sequences.Helices" title="modlamp.sequences.Helices"><code class="xref py py-class docutils literal"><span class="pre">Helices</span></code></a>,
<a class="reference internal" href="#modlamp.sequences.Kinked" title="modlamp.sequences.Kinked"><code class="xref py py-class docutils literal"><span class="pre">Kinked</span></code></a>, <a class="reference internal" href="#modlamp.sequences.Oblique" title="modlamp.sequences.Oblique"><code class="xref py py-class docutils literal"><span class="pre">Oblique</span></code></a> or <a class="reference internal" href="#modlamp.sequences.Random" title="modlamp.sequences.Random"><code class="xref py py-class docutils literal"><span class="pre">Random</span></code></a>.</p>
<dl class="method">
<dt id="modlamp.sequences.MixedLibrary.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>seqnum</em>, <em>centrosymmetric=1</em>, <em>centroasymmetric=1</em>, <em>helix=1</em>, <em>kinked=1</em>, <em>oblique=1</em>, <em>rand=1</em>, <em>randAMP=1</em>, <em>randAMPnoCM=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#MixedLibrary.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.MixedLibrary.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>initializing method of the class <a class="reference internal" href="#modlamp.sequences.MixedLibrary" title="modlamp.sequences.MixedLibrary"><code class="xref py py-class docutils literal"><span class="pre">MixedLibrary</span></code></a>. Except from <strong>number</strong>, all other parameters are
ratios of sequences of the given sequence class.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first last simple">
<li><strong>seqnum</strong> &#8211; {int} number of sequences to be generated</li>
<li><strong>centrosymmetric</strong> &#8211; ratio of symmetric centrosymmetric sequences in the library</li>
<li><strong>centroasymmetric</strong> &#8211; ratio of asymmetric centrosymmetric sequences in the library</li>
<li><strong>helix</strong> &#8211; ratio of amphipathic helical sequences in the library</li>
<li><strong>kinked</strong> &#8211; ratio of kinked amphipathic helical sequences in the library</li>
<li><strong>oblique</strong> &#8211; ratio of oblique oriented amphipathic helical sequences in the library</li>
<li><strong>rand</strong> &#8211; ratio of random sequneces in the library</li>
<li><strong>randAMP</strong> &#8211; ratio of random sequences with APD2 amino acid distribution in the library</li>
<li><strong>randAMPnoCM</strong> &#8211; ratio of random sequences with APD2 amino acid distribution without Cys and Met in the library</li>
</ul>
</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">If duplicate sequences are created, these are removed during the creation process. It is therefore quite
probable that you will not get the exact size of library that you entered as the parameter <strong>number</strong>. If you
generate a small library, it can also happen that the size is bigger than expected, because ratios are rounded.</p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.sequences.MixedLibrary.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#MixedLibrary.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.MixedLibrary.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>This method generates a virtual sequence library with the subtype ratios initialized in class <a class="reference internal" href="#modlamp.sequences.MixedLibrary" title="modlamp.sequences.MixedLibrary"><code class="xref py py-class docutils literal"><span class="pre">MixedLibrary()</span></code></a>.
All sequences are between 7 and 28 amino acids in length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">a virtual library of sequences in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>, the sub-library class names in
<code class="xref py py-attr docutils literal"><span class="pre">names</span></code>, the number of sequences generated for each class in <code class="xref py py-attr docutils literal"><span class="pre">nums</span></code> and the library size in
<code class="xref py py-attr docutils literal"><span class="pre">libsize</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">lib</span> <span class="o">=</span> <span class="n">MixedLibrary</span><span class="p">(</span><span class="mi">10000</span><span class="p">,</span><span class="n">centrosymmetric</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span><span class="n">centroasymmetric</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span><span class="n">helix</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span><span class="n">kinked</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span><span class="n">oblique</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span><span class="n">rand</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span>
<span class="go">randAMP=10,randAMPnoCM=5)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lib</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lib</span><span class="o">.</span><span class="n">libsize</span>  <span class="c1"># as duplicates were present, the library does not have the size that was sepecified</span>
<span class="go">9126</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lib</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;RHTHVAGSWYGKMPPSPQTL&#39;,&#39;MRIKLRKIPCILAC&#39;,&#39;DGINKEVKDSYGVFLK&#39;,&#39;LRLYLRLGRVWVRG&#39;,&#39;GKLFLKGGKLFLKGGKLFLKG&#39;,...]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">lib</span><span class="o">.</span><span class="n">nums</span>
<span class="go">{&#39;AMP&#39;: 2326,</span>
<span class="go">&#39;asy&#39;: 1163,</span>
<span class="go">&#39;hel&#39;: 698,</span>
<span class="go">&#39;knk&#39;: 698,</span>
<span class="go">&#39;nCM&#39;: 1163,</span>
<span class="go">&#39;obl&#39;: 465,</span>
<span class="go">&#39;ran&#39;: 2326,</span>
<span class="go">&#39;sym&#39;: 1163}</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.sequences.MixedLibrary.prune_library">
<code class="descname">prune_library</code><span class="sig-paren">(</span><em>newsize</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#MixedLibrary.prune_library"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.MixedLibrary.prune_library" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to cut down a library to the given new size.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>newsize</strong> &#8211; new desired size of the mixed library</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">adapted library with corresponding attributes (sequences, names, libsize, nums)</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.Hepahelices">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">Hepahelices</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Hepahelices"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Hepahelices" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for peptide sequences probable to form helices and include a heparin-binding-domain.</p>
<p>This class is used to construct presumed amphipathic helices that include a heparin-binding-domain (HBD)
probable to bind heparin. The HBD sequence for alpha-helices usually has the following form: <strong>XBBBXXBX</strong>
(B: basic AA; X: hydrophobic, uncharged AA, with mainly Ser &amp; Gly).</p>
<p>More on the HBD: Munoz, E. M. &amp; Linhardt, R. J. Heparin-Binding Domains in Vascular Biology. <em>Arterioscler. Thromb.
Vasc. Biol.</em> <strong>24</strong>, 1549–1557 (2004).</p>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.3.1.</span></p>
</div>
<dl class="method">
<dt id="modlamp.sequences.Hepahelices.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#Hepahelices.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.Hepahelices.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate helical sequences with class features defined in <a class="reference internal" href="#modlamp.sequences.Hepahelices" title="modlamp.sequences.Hepahelices"><code class="xref py py-class docutils literal"><span class="pre">Hepahelices()</span></code></a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">In the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>: a list of sequences including a heparin-binding-domain.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">h</span> <span class="o">=</span> <span class="n">Hepahelices</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">8</span><span class="p">,</span><span class="mi">21</span><span class="p">)</span>  <span class="c1"># minimal length: 8, maximal length: 50</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;GRLARSLKRKLNRLVRGGGRLVRGGG&#39;, &#39;IRSIRRRLSKLARSLGRGARSLGRG&#39;, &#39;RAVKRKVNKLLKGAAKVLKGAAKVLKGAAK&#39;, ... ]</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="modlamp.sequences.AMPngrams">
<em class="property">class </em><code class="descclassname">modlamp.sequences.</code><code class="descname">AMPngrams</code><span class="sig-paren">(</span><em>seqnum</em>, <em>n_min=3</em>, <em>n_max=11</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AMPngrams"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AMPngrams" title="Permalink to this definition">¶</a></dt>
<dd><p>Class for sequence generation from the most prominent ngrams (2, 3, 4grams) found in all natural AMP
sequences extracted from the <a class="reference external" href="http://aps.unmc.edu/AP/">APD3</a> (version August 2016 with 2727 sequences).
For all 2, 3 and 4grams, all possible ngrams were generated from all sequences and the top 50 most frequent
assembled into a list. Finally, leading and tailing spaces were striped and duplicates as well as ngrams containing
spaces were removed.</p>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.4.1.</span></p>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.core.ngrams_apd" title="modlamp.core.ngrams_apd"><code class="xref py py-func docutils literal"><span class="pre">modlamp.core.ngrams_apd()</span></code></a></p>
</div>
<dl class="method">
<dt id="modlamp.sequences.AMPngrams.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>seqnum</em>, <em>n_min=3</em>, <em>n_max=11</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AMPngrams.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AMPngrams.__init__" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seqnum</strong> &#8211; {int} number of sequences to be generated</li>
<li><strong>n_min</strong> &#8211; {int} minimum number of ngrams to take for sequence assembly</li>
<li><strong>n_max</strong> &#8211; {int} maximum number of ngrams to take for sequence assembly</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">s</span> <span class="o">=</span> <span class="n">AMPngrams</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">s</span><span class="o">.</span><span class="n">generate_sequences</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">s</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;LAKSLGAGKYGGGKA&#39;, &#39;KAALESCVGGGGC&#39;, &#39;GCSGKAAAAAVG&#39;, &#39;GAASCKPCGEAKGLKVCY&#39;, &#39;IGGGCKITGESCVAGLWCGESTCGCSG&#39;, ...]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">s</span><span class="o">.</span><span class="n">ngrams</span>
<span class="go">array([&#39;AGK&#39;, &#39;CKI&#39;, &#39;RR&#39;, &#39;YGGG&#39;, &#39;LSGL&#39;, &#39;RG&#39;, &#39;YGGY&#39;, &#39;PRP&#39;, &#39;LGGG&#39;, ...]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.sequences.AMPngrams.generate_sequences">
<code class="descname">generate_sequences</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/sequences.html#AMPngrams.generate_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.sequences.AMPngrams.generate_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate sequences out of APD3 ngrams stored in <code class="xref py py-attr docutils literal"><span class="pre">ngrams</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">list of sequences in <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-modlamp.database">
<span id="modlamp-database"></span><h2>2.3. modlamp.database<a class="headerlink" href="#module-modlamp.database" title="Permalink to this headline">¶</a></h2>
<p>This module incorporates functions to connect to several peptide databases. It also allows to connect to a custom
SQL database for which the configuration is given in a specified config file.</p>
<dl class="function">
<dt id="modlamp.database.query_database">
<code class="descclassname">modlamp.database.</code><code class="descname">query_database</code><span class="sig-paren">(</span><em>table</em>, <em>columns=None</em>, <em>configfile='./modlamp/data/db_config.json'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/database.html#query_database"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.database.query_database" title="Permalink to this definition">¶</a></dt>
<dd><p>This function extracts experimental results from the modlab peptide database. All data from the given table and
column names is extracted and returned.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>table</strong> &#8211; the mysql database table to be queried</li>
<li><strong>columns</strong> &#8211; a list of the column names {str} to be extracted from the table <em>default</em>: <code class="docutils literal"><span class="pre">*</span></code> (all columns)</li>
<li><strong>configfile</strong> &#8211; location of the database configuration file containing the hostname etc. for the database to
be queried.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">{numpy.array} queried data</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">query_database</span><span class="p">(</span><span class="n">table</span><span class="o">=</span><span class="s1">&#39;modlab_experiments&#39;</span><span class="p">,</span> <span class="n">columns</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;sequence&#39;</span><span class="p">,</span> <span class="s1">&#39;MCF7_activity&#39;</span><span class="p">,</span> <span class="s1">&#39;Saureus_activity&#39;</span><span class="p">])</span>
<span class="go">Password: *********</span>
<span class="go">Connecting to MySQL database...</span>
<span class="go">connection established!</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="p">[:</span><span class="mi">5</span><span class="p">]</span>
<span class="go">array([ [&#39;ILGTILGILKGL&#39;, None, 1.0],</span>
<span class="go">        [&#39;ILGTILGFLKGL&#39;, None, 1.0],</span>
<span class="go">        [&#39;ILGNILGFLKGL&#39;, None, 1.0],</span>
<span class="go">        [&#39;ILGQILGILKGL&#39;, None, 1.0],</span>
<span class="go">        [&#39;ILGHILGYLKGL&#39;, None, 1.0]], dtype=object)</span>
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If <code class="docutils literal"><span class="pre">None</span></code> or <code class="docutils literal"><span class="pre">NULL</span></code> appears as a value, this means no data was measured for this peptide and not that
activity is none (inactive).</p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.database.query_apd">
<code class="descclassname">modlamp.database.</code><code class="descname">query_apd</code><span class="sig-paren">(</span><em>ids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/database.html#query_apd"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.database.query_apd" title="Permalink to this definition">¶</a></dt>
<dd><p>A function to query sequences from the antimicrobial peptide database <a class="reference external" href="http://aps.unmc.edu/AP/">APD</a>.
If the whole database should be scraped, simply look up the latest entry ID and take a <code class="docutils literal"><span class="pre">range(1,</span> <span class="pre">'latestID')</span></code>
as function input.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>ids</strong> &#8211; {list of int} list of APD IDs to be queried from the database</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">list of peptide sequences corresponding to entered ids.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">query_apd</span><span class="p">([</span><span class="mi">15</span><span class="p">,</span> <span class="mi">16</span><span class="p">,</span> <span class="mi">18</span><span class="p">,</span> <span class="mi">19</span><span class="p">,</span> <span class="mi">20</span><span class="p">])</span>
<span class="go">[&#39;GLFDIVKKVVGALGSL&#39;, &#39;GLFDIVKKVVGAIGSL&#39;, &#39;GLFDIVKKVVGAFGSL&#39;, &#39;GLFDIAKKVIGVIGSL&#39;, &#39;GLFDIVKKIAGHIAGSI&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.database.query_camp">
<code class="descclassname">modlamp.database.</code><code class="descname">query_camp</code><span class="sig-paren">(</span><em>ids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/database.html#query_camp"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.database.query_camp" title="Permalink to this definition">¶</a></dt>
<dd><p>A function to query sequences from the antimicrobial peptide database <a class="reference external" href="http://camp.bicnirrh.res.in/">CAMP</a>.
If the whole database should be scraped, simply look up the latest entry ID and take a <code class="docutils literal"><span class="pre">range(1,</span> <span class="pre">'latestID')</span></code>
as function input.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>ids</strong> &#8211; {list of int} list of CAMP IDs to be queried from the database</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">list of peptide sequences corresponding to entered ids.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">query_camp</span><span class="p">([</span><span class="mi">2705</span><span class="p">,</span> <span class="mi">2706</span><span class="p">])</span>
<span class="go">[&#39;GLFDIVKKVVGALGSL&#39;, &#39;GLFDIVKKVVGTLAGL&#39;]</span>
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="module-modlamp.datasets">
<span id="modlamp-datasets"></span><h2>2.4. modlamp.datasets<a class="headerlink" href="#module-modlamp.datasets" title="Permalink to this headline">¶</a></h2>
<p>This module incorporates functions to load different peptide datasets used for classification.</p>
<table border="1" class="docutils">
<colgroup>
<col width="28%" />
<col width="72%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Function</th>
<th class="head">Data</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#modlamp.datasets.load_AMPvsTMset" title="modlamp.datasets.load_AMPvsTMset"><code class="xref py py-func docutils literal"><span class="pre">load_AMPvsTMset()</span></code></a></td>
<td>Antimicrobial peptides versus trans-membrane sequences</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.datasets.load_helicalAMPset" title="modlamp.datasets.load_helicalAMPset"><code class="xref py py-func docutils literal"><span class="pre">load_helicalAMPset()</span></code></a></td>
<td>Helical antimicrobial peptides versus other helical peptides</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.datasets.load_ACPvsNeg" title="modlamp.datasets.load_ACPvsNeg"><code class="xref py py-func docutils literal"><span class="pre">load_ACPvsNeg()</span></code></a></td>
<td>Helical anticancer peptides versus other mixed sequences</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.datasets.load_AMPvsUniProt" title="modlamp.datasets.load_AMPvsUniProt"><code class="xref py py-func docutils literal"><span class="pre">load_AMPvsUniProt()</span></code></a></td>
<td>AMPs from the <em>APD3</em> versus other peptides from <em>UniProt</em></td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.datasets.load_custom" title="modlamp.datasets.load_custom"><code class="xref py py-func docutils literal"><span class="pre">load_custom()</span></code></a></td>
<td>A custom data set provided in <code class="docutils literal"><span class="pre">modlamp/data</span></code> as a <code class="docutils literal"><span class="pre">.csv</span></code> file</td>
</tr>
</tbody>
</table>
<dl class="class">
<dt id="modlamp.datasets.Bunch">
<em class="property">class </em><code class="descclassname">modlamp.datasets.</code><code class="descname">Bunch</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#Bunch"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.Bunch" title="Permalink to this definition">¶</a></dt>
<dd><p>Container object for datasets</p>
<p>Dictionary-like object that exposes its keys as attributes. Taken from the <a class="reference external" href="http://scikit-learn.org">sklearn</a>
package.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">Bunch</span><span class="p">(</span><span class="n">a</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">b</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="p">[</span><span class="s1">&#39;b&#39;</span><span class="p">]</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">b</span>  <span class="c1"># key can also be called as attribute</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">a</span> <span class="o">=</span> <span class="mi">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="p">[</span><span class="s1">&#39;a&#39;</span><span class="p">]</span>
<span class="go">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">c</span> <span class="o">=</span> <span class="mi">6</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="p">[</span><span class="s1">&#39;c&#39;</span><span class="p">]</span>
<span class="go">6</span>
</pre></div>
</div>
<dl class="method">
<dt id="modlamp.datasets.Bunch.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>**kwargs</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#Bunch.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.Bunch.__init__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="function">
<dt id="modlamp.datasets.load_AMPvsTMset">
<code class="descclassname">modlamp.datasets.</code><code class="descname">load_AMPvsTMset</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#load_AMPvsTMset"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.load_AMPvsTMset" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to load a dataset consisting of <strong>AMP sequences and transmembrane regions of proteins</strong> for classification.</p>
<p>The AMP class consists of an intersection of all activity annotations of the <a class="reference external" href="http://aps.unmc.edu/AP/">APD2</a> and
<a class="reference external" href="http://camp.bicnirrh.res.in/">CAMP</a> databases, where for gram positive, gram negative and antifungal exact
matches were observed. A detailed description of how the dataset was compiled can be found in the following
publication: <em>Schneider P. et al. 2016, Mol. Inf.</em></p>
<table border="1" class="docutils">
<colgroup>
<col width="81%" />
<col width="19%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Classes</td>
<td>2</td>
</tr>
<tr class="row-even"><td>Samples per class</td>
<td>206</td>
</tr>
<tr class="row-odd"><td>Samples total</td>
<td>412</td>
</tr>
<tr class="row-even"><td>Dimensionality</td>
<td>1</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Bunch, a dictionary-like object, the interesting attributes are: <code class="docutils literal"><span class="pre">sequences</span></code>, the sequences, <code class="docutils literal"><span class="pre">target</span></code>,
the    classification labels, <code class="docutils literal"><span class="pre">target_names</span></code>, the meaning of the labels and <code class="docutils literal"><span class="pre">feature_names</span></code>, the meaning of the
features.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.datasets</span> <span class="k">import</span> <span class="n">load_AMPvsTMset</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">load_AMPvsTMset</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;AAGAATVLLVIVLLAGSYLAVLA&#39;,&#39;LWIVIACLACVGSAAALTLRA&#39;,&#39;FYRFYMLREGTAVPAVWFSIELIFGLFA&#39;,&#39;GTLELGVDYGRAN&#39;,...]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">list</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">target_names</span><span class="p">)</span>
<span class="go">[&#39;TM&#39;, &#39;AMP&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">)</span>
<span class="go">412</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">target</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span>
<span class="go">array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.datasets.load_helicalAMPset">
<code class="descclassname">modlamp.datasets.</code><code class="descname">load_helicalAMPset</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#load_helicalAMPset"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.load_helicalAMPset" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to load a dataset consisting of <strong>helical AMP sequences and other helical peptides</strong> for classification.</p>
<p>The AMP class consists of 363 helical annotated sequences from the <a class="reference external" href="http://aps.unmc.edu/AP/">APD2</a>
(extracted Dez. 14 2015). The HEL class is constructed out of extracted all alpha annotated proteins from the
<a class="reference external" href="http://www.rcsb.org/pdb/home/home.do">PDB</a>, out of which alpha helical regions were extracted. 363 sequences
were then randomly chosen from this extracted set to get equally balanced datasets in terms of numbers of sequences.</p>
<table border="1" class="docutils">
<colgroup>
<col width="81%" />
<col width="19%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Classes</td>
<td>2</td>
</tr>
<tr class="row-even"><td>Samples per class</td>
<td>363</td>
</tr>
<tr class="row-odd"><td>Samples total</td>
<td>726</td>
</tr>
<tr class="row-even"><td>Dimensionality</td>
<td>1</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Bunch, a dictionary-like object, the interesting attributes are: <code class="docutils literal"><span class="pre">sequences</span></code>, the sequences, <code class="docutils literal"><span class="pre">target</span></code>,
the classification labels, <code class="docutils literal"><span class="pre">target_names</span></code>, the meaning of the labels and <code class="docutils literal"><span class="pre">feature_names</span></code>, the meaning of the
features.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.datasets</span> <span class="k">import</span> <span class="n">load_helicalAMPset</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">load_helicalAMPset</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">[:</span><span class="mi">5</span><span class="p">]</span>
<span class="go">[&#39;FDQAQTEIQATMEEN&#39;,&#39;DVDAALHYLARLVEAG&#39;,&#39;RCPLVIDYLIDLATRS&#39;,&#39;NPATLMMFFK&#39;,&#39;NLEDSIQILRTD&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">list</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">target_names</span><span class="p">)</span>
<span class="go">[&#39;HEL&#39;, &#39;AMP&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">)</span>
<span class="go">726</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.datasets.load_ACPvsNeg">
<code class="descclassname">modlamp.datasets.</code><code class="descname">load_ACPvsNeg</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#load_ACPvsNeg"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.load_ACPvsNeg" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to load a dataset consisting of <strong>alpha-helical ACP sequences and negative peptides extracted from
alpha-helical transmembrane and non-transmembrane regions of proteins</strong> for classification.</p>
<p>The ACP class consists of a collection of ACPs from the <a class="reference external" href="http://aps.unmc.edu/AP/">APD2</a> and
<a class="reference external" href="http://crdd.osdd.net/raghava/cancerppd/index.php">CancerPPD</a> databases, manually curated by <a class="reference external" href="mailto:gisela&#46;gabernet&#37;&#52;&#48;pharma&#46;ethz&#46;ch">Gisela Gabernet</a> at modlab ETH Zürich, checking the original literature and annotated active
against at least one of the following cancer types at a concentration of 50 µM: breast, lung, skin,
haematological, and cervical. Selected sequences with length between 7 and 30 aa and without Cysteines to
facilitate synthesis.</p>
<p>The Negative peptide set contains a mixture of a random selection of 47 transmembrane alpha-helices (extracted from
the <a class="reference external" href="http://pdbtm.enzim.hu/">PDBTM</a> ) and 47 non-transmembrane helices (extracted from the <a class="reference external" href="http://www.rcsb.org/pdb/home/home.do">PDB</a>) isolated directly from the proteins crystal structure.</p>
<table border="1" class="docutils">
<colgroup>
<col width="85%" />
<col width="15%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Classes</td>
<td>2</td>
</tr>
<tr class="row-even"><td>ACP peptides</td>
<td>95</td>
</tr>
<tr class="row-odd"><td>Negative peptides</td>
<td>94</td>
</tr>
<tr class="row-even"><td>Total peptides</td>
<td>189</td>
</tr>
<tr class="row-odd"><td>Dimensionality</td>
<td>1</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Bunch, a dictionary-like object, the interesting attributes are: <code class="docutils literal"><span class="pre">sequences</span></code>, the sequences, <code class="docutils literal"><span class="pre">target</span></code>,
the classification labels, <code class="docutils literal"><span class="pre">target_names</span></code>, the meaning of the labels and <code class="docutils literal"><span class="pre">feature_names</span></code>, the meaning of the
features.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.datasets</span> <span class="k">import</span> <span class="n">load_ACPvsNeg</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">load_ACPvsNeg</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;VLTIIATIFMPLTFIAGI&#39;, &#39;QLGAGLSVGLSGLAAGFAIGIVG&#39;, &#39;WLYLILGIIFGIFGPIFNKWVL&#39;, &#39;VTWLLFLLGFVAILI&#39;...]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">list</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">target_names</span><span class="p">)</span>
<span class="go">[&#39;Neg&#39;, &#39;ACP&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">)</span>
<span class="go">189</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.datasets.load_AMPvsUniProt">
<code class="descclassname">modlamp.datasets.</code><code class="descname">load_AMPvsUniProt</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#load_AMPvsUniProt"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.load_AMPvsUniProt" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to load a dataset consisting of <strong>APD3 &lt; 80% internal similarity versus UniProt not antimicrobial,
not secretory, 10-50 AA and verified</strong> for classification.</p>
<p>The AMP class consists of 1609 AMP sequences from the <a class="reference external" href="http://aps.unmc.edu/AP/">APD3</a>. The whole APD3 was
extracted (Mar. 3rd  2016) and then submitted to an internal similarity filtering with a threshold of 80% by the
<a class="reference external" href="http://cd-hit.org">CD-Hit tool</a>. The UniProt class is constructed out of extracted protein sequences from the
<a class="reference external" href="http://uniprot.org/">UniProt Database</a> with the search query <em>NOT secretory NOT antimicrobial
AND length 10 TO 50</em>. These sequences were as well subjected to a similarity filtering of 80% yielding 4279
sequences.</p>
<table border="1" class="docutils">
<colgroup>
<col width="77%" />
<col width="23%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Classes</td>
<td>2</td>
</tr>
<tr class="row-even"><td>AMP Samples</td>
<td>1609</td>
</tr>
<tr class="row-odd"><td>UniProt Samples</td>
<td>4279</td>
</tr>
<tr class="row-even"><td>Samples total</td>
<td>5888</td>
</tr>
<tr class="row-odd"><td>Dimensionality</td>
<td>1</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">Bunch, a dictionary-like object, the interesting attributes are: <code class="docutils literal"><span class="pre">sequences</span></code>, the sequences, <code class="docutils literal"><span class="pre">target</span></code>,
the classification labels, <code class="docutils literal"><span class="pre">target_names</span></code>, the meaning of the labels and <code class="docutils literal"><span class="pre">feature_names</span></code>, the meaning of the
features.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.datasets</span> <span class="k">import</span> <span class="n">load_AMPvsUniProt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">load_AMPvsUniProt</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">[:</span><span class="mi">5</span><span class="p">]</span>
<span class="go">[&#39;GLWSKIKEVGKEAAKAAAKAAGKAALGAVSEAV&#39;,</span>
<span class="go">&#39;YVPLPNVPQPGRRPFPTFPGQGPFNPKIKWPQGY&#39;,</span>
<span class="go">&#39;DGVKLCDVPSGTWSGHCGSSSKCSQQCKDREHFAYGGACHYQFPSVKCFCKRQC&#39;,</span>
<span class="go">&#39;NLCERASLTWTGNCGNTGHCDTQCRNWESAKHGACHKRGNWKCFCYFDC&#39;,</span>
<span class="go">&#39;VFIDILDKVENAIHNAAQVGIGFAKPFEKLINPK&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">list</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">target_names</span><span class="p">)</span>
<span class="go">[&#39;AMP&#39;, &#39;UniProt&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">)</span>
<span class="go">5888</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">target</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span>
<span class="go">array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1])</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.datasets.load_custom">
<code class="descclassname">modlamp.datasets.</code><code class="descname">load_custom</code><span class="sig-paren">(</span><em>filename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/datasets.html#load_custom"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.datasets.load_custom" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to load a custom dataset saved in <code class="docutils literal"><span class="pre">modlamp/data/</span></code> as a <code class="docutils literal"><span class="pre">.csv</span></code> file.</p>
<p>The following header needs to be included: <em>Nr. of sequences</em>, <em>Nr. of columns - 1</em>, <em>Class name for 0</em>,
<em>Class name for 1</em></p>
<p>Example <code class="docutils literal"><span class="pre">.csv</span></code> file structure:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="mi">4</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">TM</span><span class="p">,</span> <span class="n">AMP</span>
<span class="n">GTLEFDVTIGRAN</span><span class="p">,</span> <span class="mi">0</span>
<span class="n">GSNVHLASNLLA</span><span class="p">,</span> <span class="mi">0</span>
<span class="n">GLFDIVKKVVGALGSL</span><span class="p">,</span> <span class="mi">0</span>
<span class="n">GLFDIIKKIAESF</span><span class="p">,</span> <span class="mi">0</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> &#8211; {str} filename of the data file to be loaded; the file must be located in <code class="docutils literal"><span class="pre">modlamp/data/</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">Bunch, a dictionary-like object, the interesting attributes are: <code class="docutils literal"><span class="pre">sequences</span></code>, the sequences, <code class="docutils literal"><span class="pre">target</span></code>,
the classification labels, <code class="docutils literal"><span class="pre">target_names</span></code>, the meaning of the labels and <code class="docutils literal"><span class="pre">feature_names</span></code>, the meaning of the
features.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.datasets</span> <span class="k">import</span> <span class="n">load_custom</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">load_custom</span><span class="p">(</span><span class="s1">&#39;custom_data.csv&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="module-modlamp.plot">
<span id="modlamp-plot"></span><h2>2.5. modlamp.plot<a class="headerlink" href="#module-modlamp.plot" title="Permalink to this headline">¶</a></h2>
<p>This module incorporates functions to plot different feature plots. The following functions are available:</p>
<table border="1" class="docutils">
<colgroup>
<col width="26%" />
<col width="74%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Function</th>
<th class="head">Characteristics</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#modlamp.plot.plot_feature" title="modlamp.plot.plot_feature"><code class="xref py py-func docutils literal"><span class="pre">plot_feature()</span></code></a></td>
<td>Generate a box plot for visualizing the distribution of a given feature.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.plot.plot_2_features" title="modlamp.plot.plot_2_features"><code class="xref py py-func docutils literal"><span class="pre">plot_2_features()</span></code></a></td>
<td>Generate a 2D scatter plot of 2 given features.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.plot.plot_3_features" title="modlamp.plot.plot_3_features"><code class="xref py py-func docutils literal"><span class="pre">plot_3_features()</span></code></a></td>
<td>Generate a 3D scatter plot of 3 given features.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.plot.plot_profile" title="modlamp.plot.plot_profile"><code class="xref py py-func docutils literal"><span class="pre">plot_profile()</span></code></a></td>
<td>Generates a profile plot of a sequence to visualize potential linear gradients.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.plot.helical_wheel" title="modlamp.plot.helical_wheel"><code class="xref py py-func docutils literal"><span class="pre">helical_wheel()</span></code></a></td>
<td>Generates a helical wheel projection plot of a given sequence.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.plot.plot_pde" title="modlamp.plot.plot_pde"><code class="xref py py-func docutils literal"><span class="pre">plot_pde()</span></code></a></td>
<td>Generates a probability density estimation plot of given data arrays.</td>
</tr>
<tr class="row-even"><td><a class="reference internal" href="#modlamp.plot.plot_violin" title="modlamp.plot.plot_violin"><code class="xref py py-func docutils literal"><span class="pre">plot_violin()</span></code></a></td>
<td>Generates a violin plot for given classes and corresponding distributions.</td>
</tr>
<tr class="row-odd"><td><a class="reference internal" href="#modlamp.plot.plot_aa_distr" title="modlamp.plot.plot_aa_distr"><code class="xref py py-func docutils literal"><span class="pre">plot_aa_distr()</span></code></a></td>
<td>Generates an amino acid frequency plot for all 20 natural amino acids.</td>
</tr>
</tbody>
</table>
<dl class="function">
<dt id="modlamp.plot.plot_feature">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_feature</code><span class="sig-paren">(</span><em>y_values</em>, <em>targets=None</em>, <em>y_label='feature values'</em>, <em>x_tick_labels=None</em>, <em>filename=None</em>, <em>colors=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_feature"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to generate a box plot of 1 given feature. The different target classes given in <strong>targets</strong> are plottet
as separate boxes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>y_values</strong> &#8211; Array of feature values to be plotted.</li>
<li><strong>targets</strong> &#8211; List of target class values [string/binary] for the given feature data.</li>
<li><strong>y_label</strong> &#8211; Axis label.</li>
<li><strong>x_tick_labels</strong> &#8211; list of labels to be assigned to the ticks on the x-axis. Must match the number of targets.</li>
<li><strong>filename</strong> &#8211; filename where to safe the plot. <em>default = None</em></li>
<li><strong>colors</strong> &#8211; {list} colors to take for plotting (strings in HEX formats).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A feature box plot.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">plot_feature</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span><span class="p">,</span><span class="n">y_label</span><span class="o">=</span><span class="s1">&#39;uH Eisenberg&#39;</span><span class="p">)</span>  <span class="c1"># d: PeptideDescriptor instance</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/uH_Eisenberg.png"><img alt="_images/uH_Eisenberg.png" src="_images/uH_Eisenberg.png" style="height: 300px;" /></a>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.plot_2_features">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_2_features</code><span class="sig-paren">(</span><em>x_values</em>, <em>y_values</em>, <em>targets=None</em>, <em>x_label=''</em>, <em>y_label=''</em>, <em>filename=None</em>, <em>colors=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_2_features"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_2_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to generate a feature scatter plot of 2 given features. The different target classes given in <strong>targets</strong>
are plottet in different colors.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>x_values</strong> &#8211; Array of values of the feature to be plotted on the x-axis.</li>
<li><strong>y_values</strong> &#8211; Array of values of the feature to be plotted on the y-axis.</li>
<li><strong>targets</strong> &#8211; List of target class values [string/binary] for the given feature data.</li>
<li><strong>x_label</strong> &#8211; X-axis label.</li>
<li><strong>y_label</strong> &#8211; Y-axis label.</li>
<li><strong>filename</strong> &#8211; filename where to safe the plot. <em>default = None</em></li>
<li><strong>colors</strong> &#8211; {list} colors to take for plotting (strings in HEX formats).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A 2D feature scatter plot.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">plot_2_features</span><span class="p">(</span><span class="n">a</span><span class="o">.</span><span class="n">descriptor</span><span class="p">,</span><span class="n">b</span><span class="o">.</span><span class="n">descriptor</span><span class="p">,</span><span class="n">x_label</span><span class="o">=</span><span class="s1">&#39;uH&#39;</span><span class="p">,</span><span class="n">y_label</span><span class="o">=</span><span class="s1">&#39;pI&#39;</span><span class="p">,</span><span class="n">targets</span><span class="o">=</span><span class="n">targs</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/2D_scatter.png"><img alt="_images/2D_scatter.png" src="_images/2D_scatter.png" style="height: 300px;" /></a>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.plot_3_features">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_3_features</code><span class="sig-paren">(</span><em>x_values</em>, <em>y_values</em>, <em>z_values</em>, <em>targets=None</em>, <em>x_label=''</em>, <em>y_label=''</em>, <em>z_label=''</em>, <em>filename=None</em>, <em>colors=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_3_features"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_3_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to generate a 3D feature scatter plot of 3 given features. The different target classes given in <strong>targets</strong>
are plottet in different colors.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>x_values</strong> &#8211; Array of values of the feature to be plotted on the x-axis.</li>
<li><strong>y_values</strong> &#8211; Array of values of the feature to be plotted on the y-axis.</li>
<li><strong>z_values</strong> &#8211; Array of values of the feature to be plotted on the z-axis.</li>
<li><strong>targets</strong> &#8211; List of target class values {string/binary} for the given feature data.</li>
<li><strong>x_label</strong> &#8211; {str} X-axis label.</li>
<li><strong>y_label</strong> &#8211; {str} Y-axis label.</li>
<li><strong>z_label</strong> &#8211; {str} Z-axis label.</li>
<li><strong>filename</strong> &#8211; {str} filename where to safe the plot. <em>default = None</em> -&gt; show the plot</li>
<li><strong>colors</strong> &#8211; {list} colors to take for plotting (strings in HEX formats).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">A 3D feature scatter plot.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">plot_3_features</span><span class="p">(</span><span class="n">a</span><span class="o">.</span><span class="n">descriptor</span><span class="p">,</span><span class="n">b</span><span class="o">.</span><span class="n">descriptor</span><span class="p">,</span><span class="n">c</span><span class="o">.</span><span class="n">descriptor</span><span class="p">,</span><span class="n">x_label</span><span class="o">=</span><span class="s1">&#39;uH&#39;</span><span class="p">,</span><span class="n">y_label</span><span class="o">=</span><span class="s1">&#39;pI&#39;</span><span class="p">,</span><span class="n">z_label</span><span class="o">=</span><span class="s1">&#39;length&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/3D_scatter.png"><img alt="_images/3D_scatter.png" src="_images/3D_scatter.png" style="height: 300px;" /></a>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.plot_profile">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_profile</code><span class="sig-paren">(</span><em>sequence</em>, <em>window=5</em>, <em>scalename='Eisenberg'</em>, <em>filename=None</em>, <em>color='red'</em>, <em>seq=False</em>, <em>ylim=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_profile"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_profile" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to generate sequence profile plots of a given amino acid scale or a moment thereof.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last"><a class="reference internal" href="#modlamp.plot.plot_profile" title="modlamp.plot.plot_profile"><code class="xref py py-func docutils literal"><span class="pre">plot_profile()</span></code></a> can only plot one-dimensional amino acid scales given in
<a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor" title="modlamp.descriptors.PeptideDescriptor"><code class="xref py py-class docutils literal"><span class="pre">modlamp.descriptors.PeptideDescriptor</span></code></a>.</p>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>sequence</strong> &#8211; {str} Peptide sequence for which the profile should be plotted.</li>
<li><strong>window</strong> &#8211; {int, uneven} Window size for which the average value is plotted for the center amino acid.</li>
<li><strong>scalename</strong> &#8211; {str} Amino acid scale to be used to describe the sequence.</li>
<li><strong>filename</strong> &#8211; {str} Filename  where to safe the plot. <em>default = None</em> &#8211;&gt; show the plot</li>
<li><strong>color</strong> &#8211; {str} Color of the plot line.</li>
<li><strong>seq</strong> &#8211; {bool} Whether the amino acid sequence should be plotted as the title.</li>
<li><strong>ylim</strong> &#8211; {tuple of float} Y-Axis limits. Provide as tuple, e.g. (0.5, -0.2)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a profile plot of the input sequence interactively or with the specified <em>filename</em></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">plot_profile</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">,</span> <span class="n">scalename</span><span class="o">=</span><span class="s1">&#39;eisenberg&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/profileplot.png"><img alt="_images/profileplot.png" src="_images/profileplot.png" style="height: 300px;" /></a>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.1.5.</span></p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.helical_wheel">
<code class="descclassname">modlamp.plot.</code><code class="descname">helical_wheel</code><span class="sig-paren">(</span><em>sequence</em>, <em>colorcoding='rainbow'</em>, <em>lineweights=True</em>, <em>filename=None</em>, <em>seq=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#helical_wheel"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.helical_wheel" title="Permalink to this definition">¶</a></dt>
<dd><p>A function to project a given peptide sequence onto a helical wheel plot. It can be useful to illustrate the
properties of alpha-helices, like positioning of charged and hydrophobic residues along the sequence.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>sequence</strong> &#8211; {str} the peptide sequence for which the helical wheel should be drawn.</li>
<li><strong>colorcoding</strong> &#8211; {str} the color coding to be used, available: <em>rainbow</em>, <em>charge</em>, <em>no</em></li>
<li><strong>lineweights</strong> &#8211; {boolean} defines whether connection lines decrease in thickness along the sequence</li>
<li><strong>filename</strong> &#8211; {str} filename  where to safe the plot. <em>default = None</em> &#8211;&gt; show the plot</li>
<li><strong>seq</strong> &#8211; {bool} whether the amino acid sequence should be plotted as a title.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a helical wheel projection plot of the given sequence (interactively or in <strong>filename</strong>)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">helical_wheel</span><span class="p">(</span><span class="s1">&#39;GLFDIVKKVVGALG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">helical_wheel</span><span class="p">(</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="n">colorcoding</span><span class="o">=</span><span class="s1">&#39;charge&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">helical_wheel</span><span class="p">(</span><span class="s1">&#39;AKLWLKAGRGFGRG&#39;</span><span class="p">,</span> <span class="n">colorcoding</span><span class="o">=</span><span class="s1">&#39;none&#39;</span><span class="p">,</span> <span class="n">lineweights</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">helical_wheel</span><span class="p">(</span><span class="s1">&#39;ACDEFGHIKLMNPQRSTVWY&#39;</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/wheel1.png"><img alt="_images/wheel1.png" src="_images/wheel1.png" style="height: 300px;" /></a>
<a class="reference internal image-reference" href="_images/wheel2.png"><img alt="_images/wheel2.png" src="_images/wheel2.png" style="height: 300px;" /></a>
<a class="reference internal image-reference" href="_images/wheel3.png"><img alt="_images/wheel3.png" src="_images/wheel3.png" style="height: 300px;" /></a>
<a class="reference internal image-reference" href="_images/wheel4.png"><img alt="_images/wheel4.png" src="_images/wheel4.png" style="height: 300px;" /></a>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.1.5.</span></p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.plot_pde">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_pde</code><span class="sig-paren">(</span><em>data</em>, <em>title=None</em>, <em>axlabels=None</em>, <em>filename=None</em>, <em>legendloc=2</em>, <em>x_min=0</em>, <em>x_max=1</em>, <em>colors=None</em>, <em>alpha=0.2</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_pde"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_pde" title="Permalink to this definition">¶</a></dt>
<dd><p>A function to plot probability density estimations of given data vectors / matrices (row wise)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data</strong> &#8211; {list / array} data of which underlying probability density function should be estimated and plotted.</li>
<li><strong>title</strong> &#8211; {str} plot title</li>
<li><strong>axlabels</strong> &#8211; {list of str} list containing the axis labels for the plot</li>
<li><strong>filename</strong> &#8211; {str} filename  where to safe the plot. <em>default = None</em> &#8211;&gt; show the plot</li>
<li><strong>legendloc</strong> &#8211; {int} location of the figures legend. 1 = top right, 2 = top left ...</li>
<li><strong>x_min</strong> &#8211; {number} x-axis minimum</li>
<li><strong>x_max</strong> &#8211; {number} x-axis maximum</li>
<li><strong>colors</strong> &#8211; {list} list of colors (readable by matplotlib, e.g. hex) to be used to plot different data classes</li>
<li><strong>alpha</strong> &#8211; {float} color alpha for filling pde curve</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">random</span><span class="p">([</span><span class="mi">3</span><span class="p">,</span><span class="mi">100</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">plot_pde</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/pde.png"><img alt="_images/pde.png" src="_images/pde.png" style="height: 300px;" /></a>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.1.</span></p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.plot_violin">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_violin</code><span class="sig-paren">(</span><em>x</em>, <em>colors=None</em>, <em>bp=False</em>, <em>filename=None</em>, <em>title=None</em>, <em>axlabels=None</em>, <em>y_min=0</em>, <em>y_max=1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_violin"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_violin" title="Permalink to this definition">¶</a></dt>
<dd><p>create violin plots out of given data array
(adapted from <a class="reference external" href="https://pyinsci.blogspot.ch/2009/09/violin-plot-with-matplotlib.html">Flavio Coelho</a>.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>x</strong> &#8211; {numpy.array} data to be plotted</li>
<li><strong>colors</strong> &#8211; {str or list} face color of the violin plots, can also be list of colors with same dimension as <strong>x</strong></li>
<li><strong>bp</strong> &#8211; {bool} print a box blot inside violin</li>
<li><strong>filename</strong> &#8211; {str} location / filename where to save the plot to. <em>default = None</em> &#8211;&gt; show the plot</li>
<li><strong>title</strong> &#8211; {str} Title of the plot.</li>
<li><strong>axlabels</strong> &#8211; {list of str} list containing the axis labels for the plot</li>
<li><strong>y_min</strong> &#8211; {number} y-axis minimum.</li>
<li><strong>y_max</strong> &#8211; {number} y-axis maximum.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">random</span><span class="o">.</span><span class="n">normal</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="p">[</span><span class="mi">5</span><span class="p">,</span> <span class="mi">100</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">plot_violin</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">colors</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;green&#39;</span><span class="p">,</span> <span class="s1">&#39;green&#39;</span><span class="p">,</span> <span class="s1">&#39;green&#39;</span><span class="p">,</span> <span class="s1">&#39;red&#39;</span><span class="p">,</span> <span class="s1">&#39;red&#39;</span><span class="p">],</span> <span class="n">bp</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/violins.png"><img alt="_images/violins.png" src="_images/violins.png" style="height: 300px;" /></a>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.2.</span></p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.plot.plot_aa_distr">
<code class="descclassname">modlamp.plot.</code><code class="descname">plot_aa_distr</code><span class="sig-paren">(</span><em>sequences</em>, <em>color='#83AF9B'</em>, <em>filename=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/plot.html#plot_aa_distr"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.plot.plot_aa_distr" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to plot the amino acid distribution of a given list of sequences</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>sequences</strong> &#8211; {list} list of sequences to calculate the amino acid distribution fore</li>
<li><strong>color</strong> &#8211; {str} color to be used (matplotlib style / hex)</li>
<li><strong>filename</strong> &#8211; {str} location / filename where to save the plot to. <em>default = None</em> &#8211;&gt; show the plot</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">plot_aa_distr</span><span class="p">([</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;WWRRWWRAARWWRRWWRR&#39;</span><span class="p">,</span> <span class="s1">&#39;ACDEFGHKLCMNPQRSTVWY&#39;</span><span class="p">,</span> <span class="s1">&#39;GGGGGIIKLWGGGGGGGGGGGGG&#39;</span><span class="p">])</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/AA_dist.png"><img alt="_images/AA_dist.png" src="_images/AA_dist.png" style="height: 300px;" /></a>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.5.</span></p>
</div>
</dd></dl>

</div>
<div class="section" id="module-modlamp.ml">
<span id="modlamp-ml"></span><h2>2.6. modlamp.ml<a class="headerlink" href="#module-modlamp.ml" title="Permalink to this headline">¶</a></h2>
<p>This module contains different functions to facilitate machine learning mainly using the scikit-learn package.
Two models are available, whose parameters can be tuned. For more information of the machine learning modules please
check the scikit-learn documentation.</p>
<table border="1" class="docutils">
<colgroup>
<col width="24%" />
<col width="76%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Model</th>
<th class="head">Reference</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Support Vector Machine</td>
<td><a class="reference external" href="http://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html">http://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html</a></td>
</tr>
<tr class="row-odd"><td>Random Forest</td>
<td><a class="reference external" href="http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html">http://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestClassifier.html</a></td>
</tr>
</tbody>
</table>
<div class="versionadded">
<p><span class="versionmodified">New in version 2.2.0.</span></p>
</div>
<dl class="function">
<dt id="modlamp.ml.train_best_model">
<code class="descclassname">modlamp.ml.</code><code class="descname">train_best_model</code><span class="sig-paren">(</span><em>model</em>, <em>x_train</em>, <em>y_train</em>, <em>scaler=StandardScaler(copy=True</em>, <em>with_mean=True</em>, <em>with_std=True)</em>, <em>score=make_scorer(matthews_corrcoef)</em>, <em>param_grid=None</em>, <em>cv=10</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/ml.html#train_best_model"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.ml.train_best_model" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns pipeline that performs standard scaling and trains
the best Support Vector Machine or Random forest classifier found by grid search.
(see sklearn.preprocessing, sklearn.grid_search, sklearn.pipeline).</p>
<p>Default parameter grids:</p>
<table border="1" class="docutils">
<colgroup>
<col width="15%" />
<col width="85%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Model</th>
<th class="head">Parameter grid</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>SVM Model</td>
<td>param_grid = [{&#8216;clf__C&#8217;: param_range, &#8216;clf__kernel&#8217;: [&#8216;linear&#8217;]},
{&#8216;clf__C&#8217;: param_range,    &#8216;clf__gamma&#8217;: param_range, &#8216;clf__kernel&#8217;: [&#8216;rbf&#8217;]}]</td>
</tr>
<tr class="row-odd"><td>Random Forest</td>
<td>param_grid = [{&#8216;clf__n_estimators&#8217;: [10, 50, 100, 500],
&#8216;clf__max_depth&#8217;: [3, None],
&#8216;clf__max_features&#8217;: [1, 2, 3, 5, 10],
&#8216;clf__min_samples_split&#8217;: [1, 3, 5, 10],
&#8216;clf__min_samples_leaf&#8217;: [1, 3, 5, 10],
&#8216;clf__bootstrap&#8217;: [True, False],
&#8216;clf__criterion&#8217;: [&#8220;gini&#8221;, &#8220;entropy&#8221;]}]</td>
</tr>
</tbody>
</table>
<p>Useful methods implemented in scikit-learn:</p>
<table border="1" class="docutils">
<colgroup>
<col width="22%" />
<col width="78%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Method</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>fit(X,y)</td>
<td>fit the model with the same parameters to new training data.</td>
</tr>
<tr class="row-odd"><td>score(X,y)</td>
<td>get the score of the model for test data.</td>
</tr>
<tr class="row-even"><td>predict(X)</td>
<td>get predictions for new data.</td>
</tr>
<tr class="row-odd"><td>predict_proba(X)</td>
<td>get probability predicitons for [class0, class1]</td>
</tr>
<tr class="row-even"><td>get_params()</td>
<td>get parameters of the trained model</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>model</strong> &#8211; {str} model to train. Choose between &#8216;svm&#8217; (Support Vector Machine) or &#8216;rf&#8217; (Random Forest).</li>
<li><strong>x_train</strong> &#8211; {array} descriptor values for training data.</li>
<li><strong>y_train</strong> &#8211; {array} class values for training data.</li>
<li><strong>scaler</strong> &#8211; {scaler} scaler to use in the pipe to scale data prior to training. Choose from sklearn.preprocessing.
E.g. StandardScaler(), MinMaxScaler(), Normalizer().</li>
<li><strong>score</strong> &#8211; {metrics instance} scoring function built from make_scorer() or a predefined value in string form
(choose from <a class="reference external" href="http://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter">http://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter</a>).</li>
<li><strong>param_grid</strong> &#8211; {dict} parameter grid for the gridsearch (see sklearn.grid_search).</li>
<li><strong>cv</strong> &#8211; {int} number of folds for cross-validation.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">best estimator pipeline.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.ml</span> <span class="k">import</span> <span class="n">train_best_model</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp.datasets</span> <span class="k">import</span> <span class="n">load_ACPvsNeg</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">modlamp</span> <span class="k">import</span> <span class="n">descriptors</span>
</pre></div>
</div>
<p>Loading a dataset for training.</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">data</span> <span class="o">=</span> <span class="n">load_ACPvsNeg</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">[</span><span class="mi">188</span><span class="p">]</span>
<span class="go">&#39;FLFKLIPKAIKGLVKAIRK&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">target</span><span class="p">[</span><span class="mi">188</span><span class="p">]</span>
<span class="go">&#39;1&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">data</span><span class="o">.</span><span class="n">target_names</span>
<span class="go">array([&#39;Neg&#39;, &#39;ACP&#39;], dtype=&#39;|S3&#39;)</span>
</pre></div>
</div>
<p>Calculating Pepcats descriptor in autocorrelation:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">descr</span> <span class="o">=</span> <span class="n">descriptors</span><span class="o">.</span><span class="n">PeptideDescriptor</span><span class="p">(</span><span class="n">data</span><span class="o">.</span><span class="n">sequences</span><span class="p">,</span><span class="n">scalename</span><span class="o">=</span><span class="s1">&#39;pepcats&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">descr</span><span class="o">.</span><span class="n">calculate_autocorr</span><span class="p">(</span><span class="mi">7</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">descr</span><span class="o">.</span><span class="n">descriptor</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="go">array([ 0.77777778,  0.11111111,  0.22222222,  0.16666667,  0.        ,</span>
<span class="go">    0.        ,  0.52941176,  0.        ,  0.        ,  0.        ,</span>
<span class="go">    0.        ,  0.        ,  0.5625    ,  0.        ,  0.        ,</span>
<span class="go">    0.        ,  0.        ,  0.        ,  0.6       ,  0.        ,</span>
<span class="go">    0.13333333,  0.        ,  0.        ,  0.        ,  0.71428571,</span>
<span class="go">    0.        ,  0.07142857,  0.07142857,  0.        ,  0.        ,</span>
<span class="go">    0.53846154,  0.07692308,  0.        ,  0.        ,  0.        ,</span>
<span class="go">    0.        ,  0.66666667,  0.        ,  0.08333333,  0.08333333,</span>
<span class="go">    0.        ,  0.        ])</span>
</pre></div>
</div>
<p>Training an SVM model with this data:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">X_train</span> <span class="o">=</span> <span class="n">descr</span><span class="o">.</span><span class="n">descriptor</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">y_train</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">target</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">best_svm_model</span> <span class="o">=</span> <span class="n">train_best_model</span><span class="p">(</span><span class="s1">&#39;svm&#39;</span><span class="p">,</span> <span class="n">X_train</span><span class="p">,</span> <span class="n">y_train</span><span class="p">)</span>
<span class="go">Best score and parameters from a 10-fold cross validation:</span>
<span class="go">mean: 0.86932, std: 0.10581, params: {&#39;clf__gamma&#39;: 0.001, &#39;clf__C&#39;: 100.0, &#39;clf__kernel&#39;: &#39;rbf&#39;}</span>
</pre></div>
</div>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">best_svm_model</span><span class="o">.</span><span class="n">get_params</span><span class="p">()</span>
<span class="go">{&#39;clf&#39;: SVC(C=100.0, cache_size=200, class_weight=&#39;balanced&#39;, coef0=0.0,</span>
<span class="go">   decision_function_shape=None, degree=3, gamma=0.001, kernel=&#39;rbf&#39;,</span>
<span class="go">   max_iter=-1, probability=True, random_state=1, shrinking=True, tol=0.001,</span>
<span class="go">   verbose=False),</span>
<span class="go"> &#39;clf__C&#39;: 100.0,</span>
<span class="go"> &#39;clf__cache_size&#39;: 200,</span>
<span class="go"> &#39;clf__class_weight&#39;: None,</span>
<span class="go"> &#39;clf__coef0&#39;: 0.0,</span>
<span class="go"> &#39;clf__decision_function_shape&#39;: None,</span>
<span class="go"> &#39;clf__degree&#39;: 3,</span>
<span class="go"> &#39;clf__gamma&#39;: 0.001,</span>
<span class="go"> &#39;clf__kernel&#39;: &#39;rbf&#39;,</span>
<span class="go"> &#39;clf__max_iter&#39;: -1,</span>
<span class="go"> &#39;clf__probability&#39;: True,</span>
<span class="go"> &#39;clf__random_state&#39;: 1,</span>
<span class="go"> &#39;clf__shrinking&#39;: True,</span>
<span class="go"> &#39;clf__tol&#39;: 0.001,</span>
<span class="go"> &#39;clf__verbose&#39;: False,</span>
<span class="go"> &#39;scl&#39;: StandardScaler(copy=True, with_mean=True, with_std=True),</span>
<span class="go"> &#39;scl__copy&#39;: True,</span>
<span class="go"> &#39;scl__with_mean&#39;: True,</span>
<span class="go"> &#39;scl__with_std&#39;: True,</span>
<span class="go"> &#39;steps&#39;: [(&#39;scl&#39;, StandardScaler(copy=True, with_mean=True, with_std=True)),</span>
<span class="go">  (&#39;clf&#39;, SVC(C=100.0, cache_size=200, class_weight=&#39;balanced&#39;, coef0=0.0,</span>
<span class="go">     decision_function_shape=None, degree=3, gamma=0.001, kernel=&#39;rbf&#39;,</span>
<span class="go">     max_iter=-1, probability=True, random_state=1, shrinking=True, tol=0.001,</span>
<span class="go">     verbose=False))]}</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.ml.plot_validation_curve">
<code class="descclassname">modlamp.ml.</code><code class="descname">plot_validation_curve</code><span class="sig-paren">(</span><em>classifier</em>, <em>x_train</em>, <em>y_train</em>, <em>param_name</em>, <em>param_range=None</em>, <em>cv=10</em>, <em>score=make_scorer(matthews_corrcoef)</em>, <em>title='Validation Curve'</em>, <em>xlab='parameter range'</em>, <em>ylab='MCC'</em>, <em>filename=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/ml.html#plot_validation_curve"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.ml.plot_validation_curve" title="Permalink to this definition">¶</a></dt>
<dd><p>Plotting cross-validation curve for the specified classifier, training data and parameter.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>classifier</strong> &#8211; {classifier instance} classifier or validation curve (e.g. sklearn.svm.SVC).</li>
<li><strong>x_train</strong> &#8211; {array} descriptor values for training data.</li>
<li><strong>y_train</strong> &#8211; {array} class values for training data.</li>
<li><strong>param_name</strong> &#8211; {string} parameter to assess in the validation curve plot. For SVM,
&#8220;clf__C&#8221; (C parameter), &#8220;clf__gamma&#8221; (gamma parameter). For RF, &#8220;clf__n_estimators&#8221; (number of trees),
&#8220;clf__max_depth&#8221; (max num of branches per tree, &#8220;clf__min_samples_split&#8221; (min number of samples required to split an
internal tree node), &#8220;clf__min_samples_leaf&#8221; (min number of samples in newly created leaf).</li>
<li><strong>param_range</strong> &#8211; {list} parameter range for the validation curve.</li>
<li><strong>cv</strong> &#8211; {int} number of folds for cross-validation.</li>
<li><strong>score</strong> &#8211; {metrics instance} scoring function built from make_scorer() or a predefined value in string form
(choose from <a class="reference external" href="http://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter">http://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter</a>).</li>
<li><strong>title</strong> &#8211; {str} graph title</li>
<li><strong>xlab</strong> &#8211; {str} x axis label.</li>
<li><strong>ylab</strong> &#8211; {str} y axis label.</li>
<li><strong>filename</strong> &#8211; {str} if filename given the figure is stored in the specified path.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">plot of the validation curve.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.ml.predict">
<code class="descclassname">modlamp.ml.</code><code class="descname">predict</code><span class="sig-paren">(</span><em>classifier</em>, <em>x_test</em>, <em>seqs_test</em>, <em>names_test=None</em>, <em>y_test=array([]</em>, <em>dtype=float64)</em>, <em>filename=None</em>, <em>save_csv=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/ml.html#predict"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.ml.predict" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns pandas dataframe with predictions using the specified estimator and test data. If true class is provided,
it returns the scoring value for the test data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>classifier</strong> &#8211; {classifier instance} classifier used for predictions.</li>
<li><strong>x_test</strong> &#8211; {array} descriptor values for testing data.</li>
<li><strong>names_test</strong> &#8211; {list} names of the peptides in test data.</li>
<li><strong>seqs_test</strong> &#8211; {list} sequences of the peptides in test data.</li>
<li><strong>y_test</strong> &#8211; {array} true classes for testing data (optional).</li>
<li><strong>filename</strong> &#8211; {string} valid path for the file to store the predictions.</li>
<li><strong>save_csv</strong> &#8211; {bool} if true additionally saves csv file with predicitons.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">pandas dataframe containing predictions for test data. Pred_prob_class0 and Pred_prob_class1
are the predicted probability of the peptide belonging to class0 and class1, respectively.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.ml.score_cv">
<code class="descclassname">modlamp.ml.</code><code class="descname">score_cv</code><span class="sig-paren">(</span><em>classifier</em>, <em>X</em>, <em>y</em>, <em>cv=10</em>, <em>metrics=None</em>, <em>names=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/ml.html#score_cv"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.ml.score_cv" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the cross validation scores for the specified scoring metrics as a pandas data frame.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>classifier</strong> &#8211; {classifier instance} trained classifier used for predictions.</li>
<li><strong>X</strong> &#8211; {array} descriptor values for training data.</li>
<li><strong>y</strong> &#8211; {array} class values for training data.</li>
<li><strong>cv</strong> &#8211; {int} number of folds for cross-validation.</li>
<li><strong>metrics</strong> &#8211; {list} metrics to consider for calculating the cv_scores. Choose from sklearn.metrics.scorers
(<a class="reference external" href="http://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter">http://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter</a>).</li>
<li><strong>names</strong> &#8211; {list} names of the metrics to display on the dataframe.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">pandas dataframe containing the cross validation scores for the specified metrics.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.ml.score_testset">
<code class="descclassname">modlamp.ml.</code><code class="descname">score_testset</code><span class="sig-paren">(</span><em>classifier</em>, <em>X_test</em>, <em>y_test</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/ml.html#score_testset"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.ml.score_testset" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the test set scores for the specified scoring metrics as a pandas data frame. The calculated metrics
are Matthews correlation coefficient, accuracy, precision, recall, f1 and Area under the Receiver-Operator curve
(roc_auc). See sklearn.metrics for more information
(<a class="reference external" href="http://scikit-learn.org/stable/modules/classes.html#sklearn-metrics-metrics">http://scikit-learn.org/stable/modules/classes.html#sklearn-metrics-metrics</a>).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>classifier</strong> &#8211; {classifier instance} trained classifier used for predictions.</li>
<li><strong>X_test</strong> &#8211; {array} descriptor values for the test data.</li>
<li><strong>y_test</strong> &#8211; {array} class values for the test data.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">pandas dataframe containing the cross validation scores for the specified metrics.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-modlamp.wetlab">
<span id="modlamp-wetlab"></span><h2>2.7. modlamp.wetlab<a class="headerlink" href="#module-modlamp.wetlab" title="Permalink to this headline">¶</a></h2>
<p>This module incorporates functions to load raw data files from wetlab experiments, calculate different
characteristics and plot.</p>
<table border="1" class="docutils">
<colgroup>
<col width="28%" />
<col width="72%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Class</th>
<th class="head">Data</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><a class="reference internal" href="#modlamp.wetlab.CD" title="modlamp.wetlab.CD"><code class="xref py py-class docutils literal"><span class="pre">CD</span></code></a></td>
<td>Class for handling Circular Dichroism data.</td>
</tr>
</tbody>
</table>
<dl class="class">
<dt id="modlamp.wetlab.CD">
<em class="property">class </em><code class="descclassname">modlamp.wetlab.</code><code class="descname">CD</code><span class="sig-paren">(</span><em>directory</em>, <em>wmin</em>, <em>wmax</em>, <em>amide=True</em>, <em>pathlen=0.1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD" title="Permalink to this definition">¶</a></dt>
<dd><p>Class to handle circular dichroism data files and calculate several ellipticity and helicity values.
The class can only handle data files of the <em>Applied Photophysics</em> type.</p>
<p>For explanations of different units used in CD spectroscopy,
visit <a class="reference external" href="https://www.photophysics.com/resources/7-cd-units-conversions">https://www.photophysics.com/resources/7-cd-units-conversions</a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">All files which should be read must have <strong>4 header lines</strong> as shown in the image below. CD data to be read
must start in line 5 (separated in 2 columns: <em>Wavelength</em> and <em>Signal</em>).</p>
</div>
<img alt="_images/fileheader.png" src="_images/fileheader.png" />
<p>First line: <em>Molecule Name</em></p>
<p>Second line: <em>Sequence</em></p>
<p>Third line: <em>concentration in µM</em></p>
<p>Fourth line: <em>solvent</em></p>
<p>Recognized solvents are <strong>W</strong> for water and <strong>T</strong> for TFE.</p>
<dl class="method">
<dt id="modlamp.wetlab.CD.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>directory</em>, <em>wmin</em>, <em>wmax</em>, <em>amide=True</em>, <em>pathlen=0.1</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Init method for class CD.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>directory</strong> &#8211; {str} directory containing all data files to be read. Files need a <strong>.csv</strong> ending</li>
<li><strong>wmin</strong> &#8211; {int} smalles wavelength measured</li>
<li><strong>wmax</strong> &#8211; {int} highest wavelength measured</li>
<li><strong>amide</strong> &#8211; {bool} specifies whether the sequences have amidated C-termini</li>
<li><strong>pathlen</strong> &#8211; {float} cuvette path length in cm</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span> <span class="o">=</span> <span class="n">CD</span><span class="p">(</span><span class="s1">&#39;/path/to/your/folder&#39;</span><span class="p">,</span> <span class="mi">185</span><span class="p">,</span> <span class="mi">260</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">filenames</span>
<span class="go">[&#39;160819_Pep1_T_smooth.csv&#39;, &#39;160819_Pep1_W_smooth.csv&#39;, &#39;160819_Pep5_T_smooth.csv&#39;, ...]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">names</span>
<span class="go">[&#39;Pep 10&#39;, &#39;Pep 10&#39;, &#39;Pep 11&#39;, &#39;Pep 11&#39;, ... ]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">conc_umol</span>
<span class="go">[33.0,  33.0,  33.0,  33.0,  33.0,  33.0,  33.0,  33.0,  33.0,  33.0,  33.0,  ... ]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">meanres_mw</span>
<span class="go">[114.29920769230768, 114.29920769230768, 111.68257692307689, 111.68257692307689, ... ]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.wetlab.CD.calc_molar_ellipticity">
<code class="descname">calc_molar_ellipticity</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD.calc_molar_ellipticity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD.calc_molar_ellipticity" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the molar ellipticity for all loaded data in <code class="xref py py-attr docutils literal"><span class="pre">circular_dichroism</span></code> according
to the following formula:</p>
<div class="math">
<p><img src="_images/math/5f2a4eea44f400fea6c5bcc1eadf9e87cdd70f20.png" alt="[\Theta] = (\Theta * MW * 100) / (c * l)"/></p>
</div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">{numpy array} molar ellipticity in <code class="xref py py-attr docutils literal"><span class="pre">molar_ellipticity</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">calc_molar_ellipticity</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">molar_ellipticity</span>
<span class="go">array([[  260.,  -1.40893636e+07],</span>
<span class="go">       [  259.,  -1.00558182e+07],</span>
<span class="go">       [  258.,  -1.25173636e+07], ...</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.wetlab.CD.calc_meanres_ellipticity">
<code class="descname">calc_meanres_ellipticity</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD.calc_meanres_ellipticity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD.calc_meanres_ellipticity" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the mean residue ellipticity for all loaded data in <code class="xref py py-attr docutils literal"><span class="pre">circular_dichroism</span></code>
according to the following formula:</p>
<div class="math">
<p><img src="_images/math/ada367cd767bc6e630d3949ac0e89f279436a09e.png" alt="[\Theta] = (\Theta * MRW) / (10 * c * l)

MRW = MW / (n - 1)"/></p>
</div><p>With <em>MRW</em> = mean residue weight, <em>n</em> = number of residues in the peptide, <em>c</em> = concentration and <em>l</em> =
cuvette path length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">{numpy array} molar ellipticity in <code class="xref py py-attr docutils literal"><span class="pre">meanres_ellipticity</span></code></td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">calc_meanres_ellipticity</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">meanres_ellipticity</span>
<span class="go">array([[   260.        ,   -266.95804196],</span>
<span class="go">       [   259.        ,   -338.13286713],</span>
<span class="go">       [   258.        ,   -387.25174825], ...</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.wetlab.CD.plot">
<code class="descname">plot</code><span class="sig-paren">(</span><em>data='mean residue ellipticity'</em>, <em>combine=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD.plot"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD.plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate CD plots for all read data in the initial directory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>data</strong> &#8211; {str} which data should be plotted (<code class="docutils literal"><span class="pre">mean</span> <span class="pre">residue</span> <span class="pre">ellipticity</span></code>, <code class="docutils literal"><span class="pre">molar</span> <span class="pre">ellipticity</span></code> or
<code class="docutils literal"><span class="pre">circular</span> <span class="pre">dichroism</span></code>)</li>
<li><strong>combine</strong> &#8211; {bool} if <code class="docutils literal"><span class="pre">True</span></code>, overlays of different solvents will be created for the same molecule.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">.pdf plots saved to the directory containing the read files.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span> <span class="o">=</span> <span class="n">CD</span><span class="p">(</span><span class="s1">&#39;/path/to/your/folder&#39;</span><span class="p">,</span> <span class="mi">180</span><span class="p">,</span> <span class="mi">260</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">calc_meanres_ellipticity</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="s1">&#39;mean residue ellipticity&#39;</span><span class="p">,</span> <span class="n">combine</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/cd1.png"><img alt="_images/cd1.png" src="_images/cd1.png" style="height: 300px;" /></a>
<a class="reference internal image-reference" href="_images/cd2.png"><img alt="_images/cd2.png" src="_images/cd2.png" style="height: 300px;" /></a>
<a class="reference internal image-reference" href="_images/cd3.png"><img alt="_images/cd3.png" src="_images/cd3.png" style="height: 300px;" /></a>
</dd></dl>

<dl class="method">
<dt id="modlamp.wetlab.CD.dichroweb">
<code class="descname">dichroweb</code><span class="sig-paren">(</span><em>data='mean residue ellipticity'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD.dichroweb"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD.dichroweb" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to save the calculated CD data into DichroWeb readable format (semi-colon separated). The produced
files can then directly be uploaded to the <a class="reference external" href="http://dichroweb.cryst.bbk.ac.uk">DichroWeb</a> analysis tool.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>data</strong> &#8211; {str} which data should be plotted (<code class="docutils literal"><span class="pre">mean</span> <span class="pre">residue</span> <span class="pre">ellipticity</span></code>, <code class="docutils literal"><span class="pre">molar</span> <span class="pre">ellipticity</span></code> or
<code class="docutils literal"><span class="pre">circular</span> <span class="pre">dichroism</span></code>)</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">.csv data files saved to the directory containing the read files.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.wetlab.CD.helicity">
<code class="descname">helicity</code><span class="sig-paren">(</span><em>temperature=24.0</em>, <em>k=3.5</em>, <em>induction=True</em>, <em>filename=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/wetlab.html#CD.helicity"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.wetlab.CD.helicity" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the percentage of helicity out of the mean residue ellipticity data.
The calculation is based on the fromula by Fairlie and co-workers:</p>
<div class="math">
<p><img src="_images/math/43cc25c938a0f091af4a4b687d5b073d7bcffdf5.png" alt="[\Theta]_{222\infty} = (-44000 * 250 * T) * (1 - k / N)"/></p>
</div><p>The helicity is then calculated as the ratio of</p>
<div class="math">
<p><img src="_images/math/c5bcef898662e2800a2ce52971f99526331a0c85.png" alt="([\Theta]_{222} / [\Theta]_{222\infty}) * 100 \%"/></p>
</div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Reference:</th><td class="field-body"><p class="first"><a class="reference external" href="https://dx.doi.org/10.1021/ja0456003">Shepherd, N. E., Hoang, H. N., Abbenante, G. &amp; Fairlie, D. P. J. Am. Chem. Soc. 127, 2974–2983
(2005).</a></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>temperature</strong> &#8211; {float} experiment temperature in °C</li>
<li><strong>k</strong> &#8211; {float, 2.4 - 4.5} finite length correction factor. Can be adapted to the helicity of a known peptide.</li>
<li><strong>induction</strong> &#8211; {bool} wether the helical induction upon changing from one solvent to another should be
calculated.</li>
<li><strong>filename</strong> &#8211; {str} if given, helicity data is saved to the file &#8220;filename&#8221;.csv</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">approximate helicity for every sequence in the attribute <code class="xref py py-attr docutils literal"><span class="pre">helicity_values</span></code>.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">calc_meanres_ellipticity</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">helicity</span><span class="p">(</span><span class="n">temperature</span><span class="o">=</span><span class="mf">24.</span><span class="p">,</span> <span class="n">k</span><span class="o">=</span><span class="mf">3.492185008</span><span class="p">,</span> <span class="n">induction</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cd</span><span class="o">.</span><span class="n">helicity_values</span>
<span class="go">            Name    Solvent   Helicity  Induction</span>
<span class="go">    0  Aurein2.2d2       T    100.0     3.823</span>
<span class="go">    1  Aurein2.2d2       W    26.16     0.000</span>
<span class="go">    2       Klak14       T    76.38     3.048</span>
<span class="go">    3       Klak14       W    25.06     0.000</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-modlamp.analysis">
<span id="modlamp-analysis"></span><h2>2.8. modlamp.analysis<a class="headerlink" href="#module-modlamp.analysis" title="Permalink to this headline">¶</a></h2>
<p>This module can be used for diverse analysis of given peptide libraries.</p>
<dl class="class">
<dt id="modlamp.analysis.GlobalAnalysis">
<em class="property">class </em><code class="descclassname">modlamp.analysis.</code><code class="descname">GlobalAnalysis</code><span class="sig-paren">(</span><em>library</em>, <em>names=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for amino acid sequence library analysis</p>
<div class="versionadded">
<p><span class="versionmodified">New in version 2.6.0.</span></p>
</div>
<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>library</em>, <em>names=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.__init__" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>library</strong> &#8211; {list, numpy.ndarray, pandas.DataFrame} sequence library, if 2D, the rows are considered as
sub-libraries.</li>
<li><strong>names</strong> &#8211; {list} list of library names to plot as labels and legend</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">g</span> <span class="o">=</span> <span class="n">GlobalAnalysis</span><span class="p">([</span><span class="s1">&#39;GLFDIVKKVVGALG&#39;</span><span class="p">,</span> <span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="o">...</span><span class="p">],</span> <span class="n">names</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;Library1&#39;</span><span class="p">])</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.calc_aa_freq">
<code class="descname">calc_aa_freq</code><span class="sig-paren">(</span><em>plot=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.calc_aa_freq"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.calc_aa_freq" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to get the frequency of every amino acid in the library. If the library consists of sub-libraries,
the frequencies of these are calculated independently.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>plot</strong> &#8211; {bool} whether the amino acid frequencies should be plotted in a histogram.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">{numpy.ndarray} amino acid frequencies in the attribute <code class="xref py py-attr docutils literal"><span class="pre">aafreq</span></code>. The values are oredered
alphabetically.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">g</span> <span class="o">=</span> <span class="n">GlobalAnalysis</span><span class="p">(</span><span class="n">sequences</span><span class="p">)</span>  <span class="c1"># sequences being a numpy array of amino acid sequences</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">g</span><span class="o">.</span><span class="n">calc_aa_freq</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">g</span><span class="o">.</span><span class="n">aafreq</span>
<span class="go">    array([[ 0.08250071,  0.        ,  0.02083928,  0.0159863 ,  0.1464459 ,</span>
<span class="go">             0.04795889,  0.06622895,  0.0262632 ,  0.12988867,  0.        ,</span>
<span class="go">             0.09192121,  0.03111619,  0.01712818,  0.04852983,  0.05937768,</span>
<span class="go">             0.07079646,  0.04396232,  0.0225521 ,  0.05994862,  0.01855552]])</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/aa_anal.png"><img alt="_images/aa_anal.png" src="_images/aa_anal.png" style="height: 300px;" /></a>
</dd></dl>

<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.calc_H">
<code class="descname">calc_H</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.calc_H"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.calc_H" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for calculating global hydrophobicity (Eisenberg scale) of all sequences in the library.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">{numpy.ndarray} Eisenberg hydrophobicities in the attribute <code class="xref py py-attr docutils literal"><span class="pre">H</span></code>.</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor.calculate_global" title="modlamp.descriptors.PeptideDescriptor.calculate_global"><code class="xref py py-func docutils literal"><span class="pre">modlamp.descriptors.PeptideDescriptor.calculate_global()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.calc_uH">
<code class="descname">calc_uH</code><span class="sig-paren">(</span><em>window=1000</em>, <em>angle=100</em>, <em>modality='max'</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.calc_uH"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.calc_uH" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for calculating hydrophobic moments (Eisenberg scale) for all sequences in the library.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>window</strong> &#8211; {int} amino acid window in which to calculate the moment. If the sequence is shorter than the
window, the length of the sequence is taken. So if the default window of 1000 is chosen, for all sequences
shorter than 1000, the <strong>global</strong> hydrophobic moment will be calculated. Otherwise, the maximal
hydrophiobic moment for the chosen window size found in the sequence will be returned.</li>
<li><strong>angle</strong> &#8211; {int} angle in which to calculate the moment. <strong>100</strong> for alpha helices, <strong>180</strong> for beta sheets.</li>
<li><strong>modality</strong> &#8211; {&#8216;max&#8217; or &#8216;mean&#8217;} calculate respectively maximum or mean hydrophobic moment.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">{numpy.ndarray} calculated hydrophobic moments in the attribute <code class="xref py py-attr docutils literal"><span class="pre">uH</span></code>.</p>
</td>
</tr>
</tbody>
</table>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last"><a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor.calculate_moment" title="modlamp.descriptors.PeptideDescriptor.calculate_moment"><code class="xref py py-func docutils literal"><span class="pre">modlamp.descriptors.PeptideDescriptor.calculate_moment()</span></code></a></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.calc_charge">
<code class="descname">calc_charge</code><span class="sig-paren">(</span><em>ph=7.0</em>, <em>amide=True</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.calc_charge"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.calc_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to calculate the total molecular charge at a given pH for all sequences in the library.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>ph</strong> &#8211; {float} ph at which to calculate the peptide charge.</li>
<li><strong>amide</strong> &#8211; {boolean} whether the sequences have an amidated C-terminus (-&gt; charge += 1).</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">{numpy.ndarray} calculated charges in the attribute <code class="xref py py-attr docutils literal"><span class="pre">charge</span></code>.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.calc_len">
<code class="descname">calc_len</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.calc_len"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.calc_len" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to get the sequence length of all sequences in the library.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">{numpy.ndarray} sequence lengths in the attribute <code class="xref py py-attr docutils literal"><span class="pre">len</span></code>.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="method">
<dt id="modlamp.analysis.GlobalAnalysis.plot_summary">
<code class="descname">plot_summary</code><span class="sig-paren">(</span><em>filename=None</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/analysis.html#GlobalAnalysis.plot_summary"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.analysis.GlobalAnalysis.plot_summary" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to generate a visual summary of different characteristics of the given library. The class methods
are used with their standard options.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>filename</strong> &#8211; {str} path to save the generated plot to.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">visual summary (plot) of the library characteristics.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">g</span> <span class="o">=</span> <span class="n">GlobalAnalysis</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([</span><span class="n">seqs1</span><span class="p">,</span> <span class="n">seqs2</span><span class="p">,</span> <span class="n">seqs3</span><span class="p">])</span>  <span class="c1"># seqs being lists of sequences</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">g</span><span class="o">.</span><span class="n">plot_summary</span><span class="p">()</span>
</pre></div>
</div>
<a class="reference internal image-reference" href="_images/summary.png"><img alt="_images/summary.png" src="_images/summary.png" style="height: 600px;" /></a>
</dd></dl>

</dd></dl>

</div>
<div class="section" id="module-modlamp.core">
<span id="modlamp-core"></span><h2>2.9. modlamp.core<a class="headerlink" href="#module-modlamp.core" title="Permalink to this headline">¶</a></h2>
<p>Core helper functions for other modules.</p>
<dl class="class">
<dt id="modlamp.core.BaseSequence">
<em class="property">class </em><code class="descclassname">modlamp.core.</code><code class="descname">BaseSequence</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Base class for sequence classes in the module <a class="reference internal" href="#module-modlamp.sequences" title="modlamp.sequences"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.sequences</span></code></a>.</p>
<dl class="method">
<dt id="modlamp.core.BaseSequence.__init__">
<code class="descname">__init__</code><span class="sig-paren">(</span><em>seqnum</em>, <em>lenmin=7</em>, <em>lenmax=28</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.__init__"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.__init__" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>seqnum</strong> &#8211; number of sequences to generate</li>
<li><strong>lenmin</strong> &#8211; minimal length of the generated sequences</li>
<li><strong>lenmax</strong> &#8211; maximal length of the generated sequences</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">attributes <code class="xref py py-attr docutils literal"><span class="pre">seqnum</span></code>, <code class="xref py py-attr docutils literal"><span class="pre">lenmin</span></code> and <code class="xref py py-attr docutils literal"><span class="pre">lenmax</span></code>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">BaseSequence</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">28</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">seqnum</span>
<span class="go">10</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">lenmin</span>
<span class="go">7</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">lenmax</span>
<span class="go">28</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.core.BaseSequence.save_fasta">
<code class="descname">save_fasta</code><span class="sig-paren">(</span><em>filename</em>, <em>names=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.save_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.save_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to save generated sequences in a <code class="docutils literal"><span class="pre">.FASTA</span></code> formatted file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> &#8211; output filename in which the sequences from <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> are safed in fasta format.</li>
<li><strong>names</strong> &#8211; {bool} whether sequence names from <code class="xref py py-attr docutils literal"><span class="pre">names</span></code> should be saved as sequence identifiers</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">a fasta file containing the generated sequences</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">BaseSequence</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;KLLSLSLALDLLS&#39;</span><span class="p">,</span> <span class="s1">&#39;KLPERTVVNSSDF&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">names</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;Sequence1&#39;</span><span class="p">,</span> <span class="s1">&#39;Sequence2&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">save_fasta</span><span class="p">(</span><span class="s1">&#39;/location/of/fasta/file.fasta&#39;</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.core.BaseSequence.mutate_AA">
<code class="descname">mutate_AA</code><span class="sig-paren">(</span><em>nr</em>, <em>prob</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.mutate_AA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.mutate_AA" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to mutate with <strong>prob</strong> probability a <strong>nr</strong> of positions per sequence randomly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>nr</strong> &#8211; number of mutations to perform per sequence</li>
<li><strong>prob</strong> &#8211; probability of mutating a sequence</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">mutated sequences in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">BaseSequence</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;IAKAGRAIIK&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">mutate_AA</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mf">1.</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;NAKAGRAWIK&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.core.BaseSequence.filter_duplicates">
<code class="descname">filter_duplicates</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.filter_duplicates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.filter_duplicates" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter duplicates in the sequences from the class attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequences list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> and corresponding names.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">BaseSequence</span><span class="p">(</span><span class="mi">4</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;KLAKLAKKLAKLAK&#39;</span><span class="p">,</span> <span class="s1">&#39;KLAKLAKKLAKLAK&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">filter_duplicates</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;KLLKLLKKLLKLLK&#39;, &#39;KLAKLAKKLAKLAK&#39;]</span>
</pre></div>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.5.</span></p>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.core.BaseSequence.keep_natural_aa">
<code class="descname">keep_natural_aa</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.keep_natural_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.keep_natural_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences that do not contain natural amino acids. If the sequence contains a character
that is not in <code class="docutils literal"><span class="pre">['A','C','D,'E','F','G','H','I','K','L','M','N','P','Q','R','S','T','V','W','Y']</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>. The other attributes are also filtered
accordingly (if present).</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">BaseSequence</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;BBBsdflUasUJfBJ&#39;</span><span class="p">,</span> <span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">keep_natural_aa</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;GLFDIVKKVVGALGSL&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.core.BaseSequence.filter_aa">
<code class="descname">filter_aa</code><span class="sig-paren">(</span><em>aminoacids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.filter_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.filter_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out corresponding names and descriptor values of sequences with given amino acids in the
argument list <em>aminoacids</em>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aminoacids</strong> &#8211; {list} amino acids to be filtered</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">filtered list of sequences names in the corresponding attributes.</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">b</span> <span class="o">=</span> <span class="n">BaseSequence</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;AAALLLIIIKKK&#39;</span><span class="p">,</span> <span class="s1">&#39;CCEERRT&#39;</span><span class="p">,</span> <span class="s1">&#39;LLVVIIFFFQQ&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">filter_aa</span><span class="p">([</span><span class="s1">&#39;C&#39;</span><span class="p">])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">b</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;AAALLLIIIKKK&#39;, &#39;LLVVIIFFFQQ&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="modlamp.core.BaseSequence.clean">
<code class="descname">clean</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#BaseSequence.clean"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.BaseSequence.clean" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to clean / clear / empty the attributes <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> and <code class="xref py py-attr docutils literal"><span class="pre">names</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">freshly initialized, empty class attributes.</td>
</tr>
</tbody>
</table>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="modlamp.core.load_scale">
<code class="descclassname">modlamp.core.</code><code class="descname">load_scale</code><span class="sig-paren">(</span><em>scalename</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#load_scale"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.load_scale" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to load scale values for a given amino acid scale</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>scalename</strong> &#8211; amino acid scale name, for available scales see the
<a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor" title="modlamp.descriptors.PeptideDescriptor"><code class="xref py py-class docutils literal"><span class="pre">modlamp.descriptors.PeptideDescriptor()</span></code></a> documentation.</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">amino acid scale values in dictionary format.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.read_fasta">
<code class="descclassname">modlamp.core.</code><code class="descname">read_fasta</code><span class="sig-paren">(</span><em>inputfile</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#read_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.read_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for loading sequences from a FASTA formatted file into <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> &amp; <code class="xref py py-attr docutils literal"><span class="pre">names</span></code>.
This method is used by the base class <a class="reference internal" href="#modlamp.descriptors.PeptideDescriptor" title="modlamp.descriptors.PeptideDescriptor"><code class="xref py py-class docutils literal"><span class="pre">modlamp.descriptors.PeptideDescriptor</span></code></a> if the input is a FASTA file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>inputfile</strong> &#8211; .fasta file with sequences and headers to read</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">list of sequences in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> with corresponding sequence names in
<code class="xref py py-attr docutils literal"><span class="pre">names</span></code>.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.save_fasta">
<code class="descclassname">modlamp.core.</code><code class="descname">save_fasta</code><span class="sig-paren">(</span><em>self</em>, <em>filename</em>, <em>names=False</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#save_fasta"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.save_fasta" title="Permalink to this definition">¶</a></dt>
<dd><p>Method for saving sequences in the instance <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> to a file in FASTA format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>filename</strong> &#8211; output filename (ending .fasta)</li>
<li><strong>names</strong> &#8211; {bool} whether sequence names from self.names should be saved as sequence identifiers</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">a FASTA formatted file containing the generated sequences</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.mutate_AA">
<code class="descclassname">modlamp.core.</code><code class="descname">mutate_AA</code><span class="sig-paren">(</span><em>self</em>, <em>nr</em>, <em>prob</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#mutate_AA"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.mutate_AA" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to mutate with <strong>prob</strong> probability a <strong>nr</strong> of positions per sequence randomly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>nr</strong> &#8211; number of mutations to perform per sequence</li>
<li><strong>prob</strong> &#8211; probability of mutating a sequence</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first">In the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>: mutated sequences</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;IAKAGRAIIK&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">mutate_AA</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="mf">1.</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;NAKAARAWIK&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.aa_weights">
<code class="descclassname">modlamp.core.</code><code class="descname">aa_weights</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#aa_weights"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.aa_weights" title="Permalink to this definition">¶</a></dt>
<dd><p>Function holding molecular weight data on all natural amino acids.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">dictionary with amino acid letters and corresponding weights</td>
</tr>
</tbody>
</table>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.4.1.</span></p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.count_aa">
<code class="descclassname">modlamp.core.</code><code class="descname">count_aa</code><span class="sig-paren">(</span><em>seq</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#count_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.count_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Function to count the amino acids occuring in a given sequence.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>seq</strong> &#8211; {str} amino acid sequence</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">{dict} dictionary with amino acids as keys and their counts in the sequence as values.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.aa_energies">
<code class="descclassname">modlamp.core.</code><code class="descname">aa_energies</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#aa_energies"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.aa_energies" title="Permalink to this definition">¶</a></dt>
<dd><p>Function holding free energies of transfer between cyclohexane and water for all natural amino acids.
H. G. Boman, D. Wade, I. a Boman, B. Wåhlin, R. B. Merrifield, <em>FEBS Lett</em>. <strong>1989</strong>, <em>259</em>, 103–106.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">dictionary with amino acid letters and corresponding energies.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.aminoacids">
<code class="descclassname">modlamp.core.</code><code class="descname">aminoacids</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#aminoacids"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.aminoacids" title="Permalink to this definition">¶</a></dt>
<dd><p>Method used by all classes in <a class="reference internal" href="#module-modlamp.sequences" title="modlamp.sequences"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.sequences</span></code></a> to generate templates for all needed instances.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">all needed instances of the classes in this package</td>
</tr>
</tbody>
</table>
<p>The following amino acid probabilities are used by modlAMP: (extracted from the
<a class="reference external" href="http://aps.unmc.edu/AP/statistic/statistic.php">APD3</a>, March 17, 2016)</p>
<table border="1" class="docutils">
<colgroup>
<col width="9%" />
<col width="13%" />
<col width="19%" />
<col width="28%" />
<col width="31%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">AA</th>
<th class="head">rand</th>
<th class="head">AMP</th>
<th class="head">AMPnoCM</th>
<th class="head">randnoCM</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>A</td>
<td>0.05</td>
<td>0.0766</td>
<td>0.0812275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>C</td>
<td>0.05</td>
<td>0.071</td>
<td>0.0</td>
<td>0.0</td>
</tr>
<tr class="row-even"><td>D</td>
<td>0.05</td>
<td>0.026</td>
<td>0.0306275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>E</td>
<td>0.05</td>
<td>0.0264</td>
<td>0.0310275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>F</td>
<td>0.05</td>
<td>0.0405</td>
<td>0.0451275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>G</td>
<td>0.05</td>
<td>0.1172</td>
<td>0.1218275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>H</td>
<td>0.05</td>
<td>0.021</td>
<td>0.0256275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>I</td>
<td>0.05</td>
<td>0.061</td>
<td>0.0656275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>K</td>
<td>0.05</td>
<td>0.0958</td>
<td>0.1004275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>L</td>
<td>0.05</td>
<td>0.0838</td>
<td>0.0884275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>M</td>
<td>0.05</td>
<td>0.0123</td>
<td>0.0</td>
<td>0.0</td>
</tr>
<tr class="row-odd"><td>N</td>
<td>0.05</td>
<td>0.0386</td>
<td>0.0432275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>P</td>
<td>0.05</td>
<td>0.0463</td>
<td>0.0509275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>Q</td>
<td>0.05</td>
<td>0.0251</td>
<td>0.0297275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>R</td>
<td>0.05</td>
<td>0.0545</td>
<td>0.0591275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>S</td>
<td>0.05</td>
<td>0.0613</td>
<td>0.0659275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>T</td>
<td>0.05</td>
<td>0.0455</td>
<td>0.0501275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>V</td>
<td>0.05</td>
<td>0.0572</td>
<td>0.0618275</td>
<td>0.05555555</td>
</tr>
<tr class="row-even"><td>W</td>
<td>0.05</td>
<td>0.0155</td>
<td>0.0201275</td>
<td>0.05555555</td>
</tr>
<tr class="row-odd"><td>Y</td>
<td>0.05</td>
<td>0.0244</td>
<td>0.0290275</td>
<td>0.05555555</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.ngrams_apd">
<code class="descclassname">modlamp.core.</code><code class="descname">ngrams_apd</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#ngrams_apd"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.ngrams_apd" title="Permalink to this definition">¶</a></dt>
<dd><p>Function returning the most frequent 2-, 3- and 4-grams from all sequences in the <a class="reference external" href="http://aps.unmc.edu/AP/">APD3</a>, version August 2016 with 2727 sequences.
For all 2, 3 and 4grams, all possible ngrams were generated from all sequences and the top 50 most frequent
assembled into a list. Finally, leading and tailing spaces were striped and duplicates as well as ngrams containing
spaces were removed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">numpy.array containing most frequent ngrams</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.template">
<code class="descclassname">modlamp.core.</code><code class="descname">template</code><span class="sig-paren">(</span><em>self</em>, <em>lenmin</em>, <em>lenmax</em>, <em>seqnum</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#template"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.template" title="Permalink to this definition">¶</a></dt>
<dd><p>Method used by different classes in <a class="reference internal" href="#module-modlamp.sequences" title="modlamp.sequences"><code class="xref py py-mod docutils literal"><span class="pre">modlamp.sequences</span></code></a> to generate length and number templates for all
needed instances.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>lenmin</strong> &#8211; minimal length of the generated sequences</li>
<li><strong>lenmax</strong> &#8211; maximal length of the generated sequences</li>
<li><strong>seqnum</strong> &#8211; number of sequences to generate</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">all needed instances (involving numbers and lengths) of the classes in this package</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.clean">
<code class="descclassname">modlamp.core.</code><code class="descname">clean</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#clean"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.clean" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to clean / clear / empty the attributes <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>, <code class="xref py py-attr docutils literal"><span class="pre">names</span></code> and <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">freshly initialized, empty class attributes.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.filter_duplicates">
<code class="descclassname">modlamp.core.</code><code class="descname">filter_duplicates</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#filter_duplicates"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.filter_duplicates" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter duplicates in the sequences from the class attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.
This function does not filter other attributes!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;DIAAVHTELVSCLNACWPALAGVCSSAL&#39;, &#39;KHVVRNKTDFFYFSQRE&#39;, &#39;WPMMVES&#39;, &#39;GHHDAPH&#39;, &#39;DIAAVHTELVSCLNACWPALAGVCSSAL&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">filter_duplicates</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;DIAAVHTELVSCLNACWPALAGVCSSAL&#39;, &#39;KHVVRNKTDFFYFSQRE&#39;, &#39;WPMMVES&#39;, &#39;GHHDAPH&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.keep_natural_aa">
<code class="descclassname">modlamp.core.</code><code class="descname">keep_natural_aa</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#keep_natural_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.keep_natural_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences that do not contain natural amino acids. If the sequence contains a character
that is not in [&#8216;A&#8217;,&#8217;C&#8217;,&#8217;D,&#8217;E&#8217;,&#8217;F&#8217;,&#8217;G&#8217;,&#8217;H&#8217;,&#8217;I&#8217;,&#8217;K&#8217;,&#8217;L&#8217;,&#8217;M&#8217;,&#8217;N&#8217;,&#8217;P&#8217;,&#8217;Q&#8217;,&#8217;R&#8217;,&#8217;S&#8217;,&#8217;T&#8217;,&#8217;V&#8217;,&#8217;W&#8217;,&#8217;Y&#8217;].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>. The other attributes are also filtered
accordingly (if present).</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.filter_unnatural">
<code class="descclassname">modlamp.core.</code><code class="descname">filter_unnatural</code><span class="sig-paren">(</span><em>self</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#filter_unnatural"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.filter_unnatural" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out sequences from the class attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code> with non-proteinogenic
amino acids [&#8216;B&#8217;, &#8216;J&#8217;, &#8216;O&#8217;, &#8216;U&#8217;, &#8216;X&#8217;, &#8216;Z&#8217;, &#8216;b, &#8216;j&#8217;, &#8216;o&#8217;, &#8216;u&#8217;, &#8216;x&#8217;, &#8216;z&#8217;].</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Returns:</th><td class="field-body">filtered sequence list in the attribute <code class="xref py py-attr docutils literal"><span class="pre">sequences</span></code>. The other attributes are also filtered
accordingly (if present).</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.filter_aa">
<code class="descclassname">modlamp.core.</code><code class="descname">filter_aa</code><span class="sig-paren">(</span><em>self</em>, <em>aminoacids</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#filter_aa"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.filter_aa" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out corresponding names and descriptor values of sequences with given amino acids in the
argument list <em>aminoacids</em>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>aminoacids</strong> &#8211; list of amino acids to be filtered</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">filtered list of sequences, descriptor values, target values and names in the corresponding attributes.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.filter_values">
<code class="descclassname">modlamp.core.</code><code class="descname">filter_values</code><span class="sig-paren">(</span><em>self</em>, <em>values</em>, <em>operator='=='</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#filter_values"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.filter_values" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter the descriptor matrix in the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code> for a given list of values (same
size as the number of features in the descriptor matrix!) The operator option tells the method whether to
filter for values equal, lower, higher ect. to the given values in the <strong>values</strong> array.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>values</strong> &#8211; List/array of values to filter the attribute <code class="xref py py-attr docutils literal"><span class="pre">descriptor</span></code> for</li>
<li><strong>operator</strong> &#8211; filter criterion, available are all SQL like operators: <code class="docutils literal"><span class="pre">==</span></code>, <code class="docutils literal"><span class="pre">&lt;</span></code>, <code class="docutils literal"><span class="pre">&gt;</span></code>, <code class="docutils literal"><span class="pre">&lt;=``and</span> <span class="pre">``&gt;=</span></code>.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">descriptor matrix and updated sequences containing only entries with descriptor values given in
<strong>values</strong> in the corresponding attributes.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.filter_sequences">
<code class="descclassname">modlamp.core.</code><code class="descname">filter_sequences</code><span class="sig-paren">(</span><em>self</em>, <em>sequences</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#filter_sequences"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.filter_sequences" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to filter out entries for given sequences in <strong>sequences</strong> out of a descriptor instance. All
corresponding fields of these sequences (<em>descriptor</em>, <em>name</em>) are deleted as well. The method returns an updated
descriptor instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>sequences</strong> &#8211; list of sequences to be filtered out of the whole instance, including corresponding data</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">updated instance without filtered sequences</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sequences</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">,</span> <span class="s1">&#39;ACDEFGHIK&#39;</span><span class="p">,</span> <span class="s1">&#39;GLFDIVKKVV&#39;</span><span class="p">,</span> <span class="s1">&#39;GLFDIVKKVVGALG&#39;</span><span class="p">,</span> <span class="s1">&#39;GLFDIVKKVVGALGSL&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="n">sequences</span><span class="p">,</span> <span class="s1">&#39;pepcats&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">calculate_crosscorr</span><span class="p">(</span><span class="mi">7</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">filter_sequences</span><span class="p">(</span><span class="s1">&#39;KLLKLLKKLLKLLK&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">sequences</span>
<span class="go">[&#39;ACDEFGHIK&#39;, &#39;GLFDIVKKVV&#39;, &#39;GLFDIVKKVVGALG&#39;, &#39;GLFDIVKKVVGALGSL&#39;]</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.random_selection">
<code class="descclassname">modlamp.core.</code><code class="descname">random_selection</code><span class="sig-paren">(</span><em>self</em>, <em>num</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#random_selection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.random_selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to randomly select a specified number of sequences (with names and descriptors if present) out of a given
descriptor instance.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><strong>num</strong> &#8211; {int} number of entries to be randomly selected</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body">updated instance</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">h</span> <span class="o">=</span> <span class="n">Helices</span><span class="p">(</span><span class="mi">7</span><span class="p">,</span> <span class="mi">28</span><span class="p">,</span> <span class="mi">100</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">h</span><span class="o">.</span><span class="n">generate_helices</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span> <span class="o">=</span> <span class="n">PeptideDescriptor</span><span class="p">(</span><span class="n">h</span><span class="o">.</span><span class="n">sequences</span><span class="p">,</span> <span class="s1">&#39;eisenberg&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">calculate_moment</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">sequences</span><span class="p">)</span>
<span class="go">100</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">100</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">d</span><span class="o">.</span><span class="n">random_selection</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">10</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">d</span><span class="o">.</span><span class="n">descriptor</span><span class="p">)</span>
<span class="go">10</span>
</pre></div>
</div>
<div class="versionadded">
<p><span class="versionmodified">New in version v2.2.3.</span></p>
</div>
</dd></dl>

<dl class="function">
<dt id="modlamp.core.minmax_selection">
<code class="descclassname">modlamp.core.</code><code class="descname">minmax_selection</code><span class="sig-paren">(</span><em>self</em>, <em>iterations</em>, <em>distmetric='euclidean'</em>, <em>randseed=0</em><span class="sig-paren">)</span><a class="reference internal" href="_modules/modlamp/core.html#minmax_selection"><span class="viewcode-link">[source]</span></a><a class="headerlink" href="#modlamp.core.minmax_selection" title="Permalink to this definition">¶</a></dt>
<dd><p>Method to select a specified number of sequences according to the minmax algorithm.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Parameters:</th><td class="field-body"><ul class="first simple">
<li><strong>iterations</strong> &#8211; {int} Number of sequences to retrieve.</li>
<li><strong>distmetric</strong> &#8211; Distance metric to calculate the distances between the sequences in descriptor space.
Choose:// docs.scipy.org / doc / scipy / reference / spatial.distance.html).
E.g. &#8216;euclidean&#8217;, &#8216;minkowsky&#8217;.</li>
<li><strong>randseed</strong> &#8211; {int} Set a random seed for numpy to pick the first sequence.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Returns:</th><td class="field-body"><p class="first last">updated instance</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/modlab.png" alt="Logo"/>
            </a></p>
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">2. Package Documentation</a><ul>
<li><a class="reference internal" href="#module-modlamp.descriptors">2.1. modlamp.descriptors</a></li>
<li><a class="reference internal" href="#module-modlamp.sequences">2.2. modlamp.sequences</a></li>
<li><a class="reference internal" href="#module-modlamp.database">2.3. modlamp.database</a></li>
<li><a class="reference internal" href="#module-modlamp.datasets">2.4. modlamp.datasets</a></li>
<li><a class="reference internal" href="#module-modlamp.plot">2.5. modlamp.plot</a></li>
<li><a class="reference internal" href="#module-modlamp.ml">2.6. modlamp.ml</a></li>
<li><a class="reference internal" href="#module-modlamp.wetlab">2.7. modlamp.wetlab</a></li>
<li><a class="reference internal" href="#module-modlamp.analysis">2.8. modlamp.analysis</a></li>
<li><a class="reference internal" href="#module-modlamp.core">2.9. modlamp.core</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="readme.html"
                        title="previous chapter">1. README</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="examplescript.html"
                        title="next chapter">3. Example scripts</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/modlamp.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="examplescript.html" title="3. Example scripts"
             >next</a> |</li>
        <li class="right" >
          <a href="readme.html" title="1. README"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">modlAMP v2.6.0 documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2016, modlab ETH Zurich, Alex Mueller.
      Last updated on Nov 28, 2016.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.4.8.
    </div>
  </body>
</html>